{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "b591ac00-eda8-4881-96f9-9e87dc2dc9a3",
   "metadata": {},
   "source": [
    "purpose: compare the manhattan plots for nicsa traits to determine what the optimal trait to use is"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e6638ee0-f7a0-4f46-adbf-0c479aa72c5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import statsmodels.stats.multitest\n",
    "import statsmodels.api as sm \n",
    "import pylab as py \n",
    "import statsmodels.api as sm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "37f8c0f1-49b8-4535-8ac8-40485f0e5b78",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('/tscc/projects/ps-palmer/brittany/rare_common_alcohol/rare_common_alcohol_comparison/notebooks/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "fb15024f-e4ab-42c5-b7c3-e24d199e7952",
   "metadata": {},
   "outputs": [],
   "source": [
    "from rca_functions import manhattan\n",
    "from rca_functions import porcupine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f8e5f2d7-8e7e-41d2-bc7d-6e4789b84e94",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('/tscc/projects/ps-palmer/brittany/SUD_cross_species/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0f2436be-d18e-4774-ab5c-e50553b89d3c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<function rca_functions.porcupine(pval, test, pos, chr, label, cut_SKAT=5e-08, cut_SKATO=5e-08, cut_burden=5e-08, chrs_plot=None, chrs_names=None, cut=2, colors=['k', '0.5'], title='Title', xlabel='chromosome', ylabel='-log10(p-value)', top=0, lines=[10, 15], lines_colors=['g', 'r'], lines_styles=['-', '--'], lines_widths=[1, 1], zoom=None, scaling='-log10', plot_grid_lines=True, **kwargs)>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def porcupine(pval, test, pos, chr, label,\n",
    "              cut_SKAT=5e-8,\n",
    "              cut_SKATO=5e-8,\n",
    "              cut_burden=5e-8,\n",
    "              chrs_plot=None, chrs_names=None,\n",
    "              cut=2,\n",
    "              colors=['k', '0.5'],\n",
    "              title='Title',\n",
    "              xlabel='chromosome',\n",
    "              ylabel='-log10(p-value)',\n",
    "              top=0,\n",
    "              lines=[10, 15],\n",
    "              lines_colors=['g', 'r'],\n",
    "              lines_styles=['-', '--'],\n",
    "              lines_widths=[1, 1],\n",
    "              zoom=None,\n",
    "              scaling='-log10',\n",
    "              plot_grid_lines=True,\n",
    "              **kwargs):\n",
    "    \"\"\"\n",
    "    Generates a specialized Porcupine plot for different types of rare-variant SNP to gene tests (SKAT, SKATO, Burden), highlighting significant findings in genomic data using color-coding for test types.\n",
    "\n",
    "    Parameters:\n",
    "    - pval (array-like): Array of p-values for genomic variants.\n",
    "    - test (array-like): Array specifying the type of genetic test performed for each variant p-value.\n",
    "    - pos (array-like): Array of positions of variants on their respective chromosomes.\n",
    "    - chr (array-like): Array of chromosome numbers for each variant.\n",
    "    - label (str): Label for the dataset, used for annotations.\n",
    "    - cut_SKAT, cut_SKATO, cut_burden (float): Cutoff p-values for SKAT, SKATO, and Burden tests, respectively.\n",
    "    - chrs_plot (list, optional): Specific chromosomes to include in the plot.\n",
    "    - chrs_names (list, optional): Custom names for the chromosomes to be plotted.\n",
    "    - cut (float): Cutoff for -log10(p-value) for displaying points on the plot.\n",
    "    - colors (list): Colors to use for plotting points, cycling through for different chromosomes.\n",
    "    - title (str): Title of the plot.\n",
    "    - xlabel, ylabel (str): Labels for the x-axis and y-axis.\n",
    "    - top (float): Upper limit for the y-axis; if 0, it is calculated from the data.\n",
    "    - lines (list): y-values where horizontal lines should be drawn.\n",
    "    - lines_colors, lines_styles, lines_widths (list): Properties for the horizontal lines.\n",
    "    - zoom (tuple): Tuple (chromosome, center position, range) for focusing on a specific region.\n",
    "    - scaling (str): P-value scaling method; supports '-log10' for negative log transformation or 'none'.\n",
    "    - plot_grid_lines (bool): Whether to include grid lines on the plot.\n",
    "    - **kwargs: Additional keyword arguments for matplotlib plot functions.\n",
    "\n",
    "    Returns:\n",
    "    matplotlib.pyplot: Configured plot object ready for display or saving.\n",
    "    \"\"\"\n",
    "    \n",
    "    # Initialize plot settings and clear any existing figures\n",
    "    shift = np.array([0.0])\n",
    "    plt.clf()\n",
    "\n",
    "    # Determine which chromosomes to plot, sorting naturally if needed\n",
    "    if chrs_plot is None:\n",
    "        chrs_list = np.unique(chr)\n",
    "        chrs_list = sorted_nicely(chrs_list) if isinstance(chrs_list[0], str) else chrs_list.sort()\n",
    "    else:\n",
    "        chrs_list = chrs_plot\n",
    "\n",
    "    # Generate chromosome labels if not provided\n",
    "    if chrs_names is None:\n",
    "        chrs_names = [str(chrs_list[i]) for i in range(len(chrs_list))]\n",
    "\n",
    "    plot_positions = len(chrs_list) == 1\n",
    "\n",
    "    # Convert cutoffs to the appropriate scale if necessary\n",
    "    if scaling == '-log10':\n",
    "        cut_burden = -np.log10(cut_burden)\n",
    "        cut_SKATO = -np.log10(cut_SKATO)\n",
    "        cut_SKAT = -np.log10(cut_SKAT)\n",
    "        \n",
    "    # Plot data for each chromosome\n",
    "    for ii, i in enumerate(chrs_list):     \n",
    "        plt.subplot(1,1,1)\n",
    "        filt = np.where(chr == i)[0]\n",
    "        x = shift[-1] + pos[filt]\n",
    "        y = -np.log10(pval[filt]) if scaling == '-log10' else pval[filt]\n",
    "        test_filter = test[filt]\n",
    "        \n",
    "        # Plot data points above a general cut-off, with specific colors for each test type. Requires the color_dict to function, or must redefine the color dictionary\n",
    "        plt.plot(x[y > cut], y[y > cut], '.', color=colors[ii % len(colors)], **kwargs)\n",
    "        plt.plot(x[(y > cut) & (test_filter == 'Burden') & (y > cut_burden)], y[(y > cut) & (test_filter == 'Burden') & (y > cut_burden)], '.', color=color_dict['Burden'], **kwargs)\n",
    "        plt.plot(x[(y > cut) & (test_filter == 'SKATO') & (y > cut_SKATO)], y[(y > cut) & (test_filter == 'SKATO') & (y > cut_SKATO)], '.', color=color_dict['SKAT-O'], **kwargs)\n",
    "        plt.plot(x[(y > cut) & (test_filter == 'SKAT') & (y > cut_SKAT)], y[(y > cut) & (test_filter == 'SKAT') & (y > cut_SKAT)], '.', color=color_dict['SKAT'], **kwargs)\n",
    "\n",
    "        # Calculate the maximum shift for the next set of points\n",
    "        shift_f = np.max(x)\n",
    "        shift_m = 0  # Placeholder for potential future use\n",
    "        shift = np.append(shift, np.max([shift_f, shift_m]))\n",
    "\n",
    "        # Set grid lines and limits\n",
    "        if plot_grid_lines:\n",
    "            plt.plot([shift[-1], shift[-1]], [0, 1000], '-', lw=0.5, color='lightgray', **kwargs)\n",
    "        plt.xlim([0, shift[-1]])\n",
    "\n",
    "    # Determine the upper limit for the y-axis\n",
    "    if top == 0:\n",
    "        top = np.ceil(np.max(-np.log10(pval))) if scaling == '-log10' else np.ceil(np.max(pval))\n",
    "\n",
    "    # Configure fig labels and horizontal lines\n",
    "    shift_label = shift[-1]\n",
    "    shift = (shift[1:] + shift[:-1]) / 2\n",
    "    for i, line_height in enumerate(lines):\n",
    "        plt.axhline(y=line_height, color=lines_colors[i], linestyle=lines_styles[i], linewidth=lines_widths[i])\n",
    "    \n",
    "    plt.ylim([cut, top])\n",
    "    plt.title(title)\n",
    "    if not plot_positions:\n",
    "        plt.xticks(shift, chrs_names)\n",
    "    plt.text(shift_label * 0.95, top * 0.95, label, verticalalignment='top', horizontalalignment='right')\n",
    "    plt.ylabel(ylabel)\n",
    "    plt.xlabel(xlabel)\n",
    "\n",
    "    # Apply zoom settings if specified\n",
    "    if zoom is not None:\n",
    "        plt.xlim([zoom_shift - zoom[2], zoom_shift + zoom[2]])\n",
    "\n",
    "    return plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "043c9b1d-7225-4e86-b1d0-b9b2350b9d22",
   "metadata": {},
   "outputs": [],
   "source": [
    "    def porcupineplotv2(self, qtltable = '', traitlist: list = [], display_figure = False, skip_manhattan = False, maxtraits = 60):\n",
    "        printwithlog('starting porcupine plot v2')\n",
    "        hv.opts.defaults(hv.opts.Points(width=1200, height=600), hv.opts.RGB(width=1200, height=600) )\n",
    "        if type(qtltable) == str:\n",
    "            if not len(qtltable): qtltable = pd.read_csv(f'{self.path}results/qtls/finalqtl.csv').reset_index().query('QTL == True')\n",
    "        if not len(traitlist): traitlist = list(map(lambda x:x.replace('regressedlr_', ''),self.traits))        \n",
    "        cmap = sns.color_palette(\"tab20\", len(traitlist))\n",
    "        d = {t: cmap[v] for v,t in enumerate(sorted(traitlist))}\n",
    "        d_inv = {cmap[v]:t for v,t in enumerate(sorted(traitlist))}\n",
    "        tnum = {t:num for num,t in enumerate(sorted(traitlist))}    \n",
    "        qtltable['color'] =  qtltable.trait.apply(lambda x: d[x]) \n",
    "        qtltable['traitnum'] =  qtltable.trait.apply(lambda x: f'{tnum[x]}') \n",
    "        if len(traitlist) > maxtraits: \n",
    "            traitlist_new = list(qtltable.trait.unique())\n",
    "            if maxtraits - len(traitlist_new) > 0:\n",
    "                traitlist_new += list(np.random.choice(list(set(traitlist) - set(traitlist_new)), maxtraits - len(traitlist_new), replace = False))\n",
    "        else: traitlist_new = traitlist\n",
    "        fdf = []\n",
    "        h2file = pd.read_csv(f'{self.path}results/heritability/heritability.tsv', sep = '\\t', index_col = 0).rename(lambda x: x.replace('regressedlr_', ''))\n",
    "        for num, t in tqdm(list(enumerate(traitlist))):\n",
    "            if not skip_manhattan or t in traitlist_new:\n",
    "                df_gwas = []\n",
    "                for opt in [f'regressedlr_{t.replace(\"regressedlr_\", \"\")}.loco.mlma', \n",
    "                            f'regressedlr_{t.replace(\"regressedlr_\", \"\")}.mlma']+ \\\n",
    "                           [f'regressedlr_{t.replace(\"regressedlr_\", \"\")}_chrgwas{chromp2}.mlma' for chromp2 in self.chrList()]:\n",
    "                    if glob(f'{self.path}results/gwas/{opt}'):\n",
    "                        df_gwas += [pd.read_csv(f'{self.path}results/gwas/{opt}', sep = '\\t')]\n",
    "                    else:  pass\n",
    "                if len(df_gwas) == 0 :  printwithlog(f'could not open mlma files for {t}')\n",
    "                df_gwas = pd.concat(df_gwas)\n",
    "                append_position = df_gwas.groupby('Chr').bp.agg('max').sort_index().cumsum().shift(1,fill_value=0)\n",
    "                qtltable['x'] = qtltable.apply(lambda x: x.bp +  append_position[x.Chr], axis = 1)\n",
    "                df_gwas['-log10p'] = -np.log10(df_gwas.p)\n",
    "                df_gwas.drop(['A1', 'A2', 'Freq', 'b', 'se', 'p'], axis = 1, inplace = True)\n",
    "                def mapcolor(c): \n",
    "                    if int(str(c).replace('X',str(self.n_autosome+1)).replace('Y', str(self.n_autosome+2)).replace('MT', str(self.n_autosome+4)))%2 == 0: return 'black'\n",
    "                    return 'gray'\n",
    "                df_gwas = df_gwas.groupby('Chr') \\\n",
    "                                 .apply(lambda df: df.assign(color = mapcolor(df.Chr[0]), x = df.bp + append_position[df.Chr[0]])) \\\n",
    "                                 .reset_index(drop = True)\n",
    "                df_gwas.loc[df_gwas['-log10p']> self.threshold, 'color' ] = str(d[t])[1:-1]\n",
    "                df_gwas.loc[df_gwas['-log10p']> self.threshold, 'color' ] = df_gwas.loc[df_gwas['-log10p']> self.threshold, 'color' ].str.split(',').map(lambda x: tuple(map(float, x)))\n",
    "                if not skip_manhattan:\n",
    "                    yrange = (-.05,max(6, df_gwas['-log10p'].max()+.5))\n",
    "                    xrange = tuple(df_gwas.x.agg(['min', 'max'])+ np.array([-1e7,+1e7]))\n",
    "                    fig = []\n",
    "                    for idx, dfs in df_gwas[df_gwas.color.isin(['gray', 'black'])].groupby('color'):\n",
    "                        temp = datashade(hv.Points(dfs, kdims = ['x','-log10p']), pixel_ratio= 2, aggregator=ds.count(), width = 1200,height = 600, y_range= yrange,\n",
    "                                 min_alpha=.7, cmap = [idx], dynamic = False )\n",
    "                        temp = dynspread(temp, max_px=4,threshold= 1 )\n",
    "                        fig += [temp]\n",
    "                    fig = fig[0]*fig[1]\n",
    "                    fig = fig*hv.HLine((self.threshold05)).opts(color='blue')*hv.HLine(self.threshold).opts(color='red')\n",
    "                    fig = fig*hv.Points(df_gwas[df_gwas['-log10p']> self.threshold].drop('color', axis = 1), \n",
    "                                        kdims = ['x','-log10p']).opts(color = 'red', size = 5)\n",
    "                    figh2 = round(h2file.loc[t.replace(\"regressedlr_\", \"\"),'V(G)/Vp'],3)\n",
    "                    fig = fig.opts(xticks=[((dfs.x.agg(['min', 'max'])).sum()//2 , self.replacenumstoXYMT(names)) for names,dfs in  df_gwas.groupby('Chr')],\n",
    "                                                   xlim =xrange, ylim=yrange, width = 1200,height = 600,  xlabel='Chromosome',\n",
    "                                   title = f'{t.replace(\"regressedlr_\", \"\")} n={self.df[\"regressedlr_\"+ t.replace(\"regressedlr_\", \"\")].count()} h2={figh2}') \n",
    "                    hv.save(fig, f'{self.path}images/manhattan/{t.replace(\"regressedlr_\", \"\")}.png')\n",
    "                if t in traitlist_new: fdf += [df_gwas]\n",
    "        fdf = pd.concat(fdf).reset_index(drop = True).sort_values('x')\n",
    "        fig = []\n",
    "        yrange = (-.05,max(6, fdf['-log10p'].max()+.5))\n",
    "        xrange = tuple(fdf.x.agg(['min', 'max'])+ np.array([-1e7,+1e7]))\n",
    "        for idx, dfs in fdf[fdf.color.isin(['gray', 'black'])].groupby('color'):\n",
    "            temp = datashade(hv.Points(dfs, kdims = ['x','-log10p']), pixel_ratio= 2, aggregator=ds.count(), width = 1200,height = 600, y_range= yrange,\n",
    "                     min_alpha=.7, cmap = [idx], dynamic = False )\n",
    "            temp = dynspread(temp, max_px=4,threshold= 1 )\n",
    "            fig += [temp]\n",
    "        fig = fig[0]*fig[1]\n",
    "        \n",
    "        fig = fig*hv.HLine((self.threshold05)).opts(color='blue')\n",
    "        fig = fig*hv.HLine(self.threshold).opts(color='red')\n",
    "        \n",
    "        for idx, dfs in fdf[~fdf.color.isin(['gray', 'black'])].groupby('color'):\n",
    "            fig = fig*hv.Points(dfs.drop('color', axis = 1), kdims = ['x','-log10p']).opts(color = idx, size = 5)\n",
    "        \n",
    "        for t, dfs in qtltable.groupby('trait'):\n",
    "            fig = fig*hv.Points(dfs.assign(**{'-log10p': qtltable.p}), kdims = ['x','-log10p'],vdims=[ 'trait','SNP' ,'A1','A2','Freq' ,'b','traitnum'], label = f'({tnum[t]}) {t}' ) \\\n",
    "                                          .opts(size = 17, color = d[t], marker='inverted_triangle', line_color = 'black', tools=['hover']) #\n",
    "        fig = fig*hv.Labels(qtltable.rename({'p':'-log10p'}, axis = 1)[['x', '-log10p', 'traitnum']], \n",
    "                            ['x','-log10p'],vdims=['traitnum']).opts(text_font_size='5pt', text_color='black')\n",
    "        fig.opts(xticks=[((dfs.x.agg(['min', 'max'])).sum()//2 , self.replacenumstoXYMT(names)) for names, dfs in fdf.groupby('Chr')],\n",
    "                                   xlim =xrange, ylim=yrange, xlabel='Chromosome', shared_axes=False,\n",
    "                               width=1200, height=600, title = f'porcupineplot',legend_position='right',show_legend=True)\n",
    "        hv.save(fig, f'{self.path}images/porcupineplot.png')\n",
    "        if display_figure: \n",
    "            display(fig)\n",
    "            return\n",
    "        return fig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5102a654-fdda-45e2-8f23-793090ddf572",
   "metadata": {},
   "outputs": [],
   "source": [
    "    def prune_genotypes(self):\n",
    "        printwithlog('starting genotype prunner...')\n",
    "        snps,_,gens = pandas_plink.read_plink(self.genotypes_subset)\n",
    "        gens = da.nan_to_num(gens, -1).astype(np.int8)\n",
    "        def prune_dups(array):\n",
    "            dict = defaultdict(list, {})\n",
    "            for num, i in enumerate(array): dict[i.tobytes()] += [num]\n",
    "            return dict\n",
    "        printwithlog('starting genotype dups finder...')    \n",
    "        pruned = prune_dups(gens.compute())\n",
    "        first_snps = [snps.loc[v[0], 'snp'] for k,v in pruned.items()]\n",
    "        printwithlog(f'saving resulst to:\\n1){self.path}pvalthresh/genomaping.parquet.gz\\n2){self.path}pvalthresh/prunned_dup_snps.in\\n3){self.path}genotypes/prunedgenotypes')  \n",
    "        prunedset = pd.DataFrame([[k,'|'.join(map(str, v))] for k,v in pruned.items()], columns = ['genotypes', 'snps'])\n",
    "        prunedset.to_parquet(f'{self.path}pvalthresh/genomaping.parquet.gz', compression = 'gzip')\n",
    "        pd.DataFrame(first_snps).to_csv(f'{self.path}pvalthresh/prunned_dup_snps.in', index = False, header = None)\n",
    "        plink(bfile=self.genotypes_subset,  thread_num = self.threadnum, extract = f'{self.path}pvalthresh/prunned_dup_snps.in',\n",
    "              make_bed = '', out = f'{self.path}genotypes/prunedgenotypes')\n",
    "        printwithlog(f'prunned data has {format(prunedset.shape[0], \",\")} out of the original {format(gens.shape[0], \",\")}')\n",
    "        return f'saved prunned genotypes to {self.path}genotypes/prunedgenotypes'\n",
    "        \n",
    "    def estimate_pval_threshold(self, replicates = 1000, sample_size = 'all', exact_prunner = True ,prunning_window = 5000000, prunning_step = 1000, remove_after = False):\n",
    "        printwithlog('starting P-value threshold calculation...')\n",
    "        if sample_size == 'all': sample_size = len(self.df)\n",
    "        if sample_size < 1: round(len(self.df)*sample_size)\n",
    "        cline = Client( processes = False) if not client._get_global_client() else client._get_global_client()\n",
    "        os.makedirs(f'{self.path}pvalthresh', exist_ok = True)\n",
    "        os.makedirs(f'{self.path}pvalthresh/gwas/', exist_ok = True)\n",
    "        os.makedirs(f'{self.path}pvalthresh/randomtrait/', exist_ok = True)\n",
    "        \n",
    "        if exact_prunner: self.prune_genotypes()\n",
    "        if not exact_prunner:\n",
    "            prunning_params = f'{prunning_window} {prunning_step} 0.999'\n",
    "            printwithlog(f'prunning gentoypes using {prunning_params}')\n",
    "            if not os.path.exists(f'{self.path}pvalthresh/pruned_data.prune.in'):\n",
    "                plink(bfile=self.genotypes_subset, indep_pairwise = prunning_params, out = f'{self.path}pvalthresh/pruned_data', thread_num = self.threadnum)\n",
    "                plink(bfile=self.genotypes_subset,  thread_num = self.threadnum, extract = f'{self.path}pvalthresh/pruned_data.prune.in', \n",
    "                      make_bed = '', out = f'{self.path}genotypes/prunedgenotypes')\n",
    "            npruned_snps = [pd.read_csv(f'{self.path}pvalthresh/pruned_data.prune.{i}', header = None).shape[0] for i in ['in', 'out']]\n",
    "            display(f'prunned data has {npruned_snps[0]} out of the original {npruned_snps[1]}' )\n",
    "        \n",
    "        def get_maxp_1sample(ranid, skip_already_present = True, remove_after = True ):\n",
    "            os.makedirs(f'{self.path}pvalthresh/gwas/{ranid}', exist_ok = True)\n",
    "            r = np.random.RandomState(ranid)\n",
    "            valuelis = r.normal(size = self.df.shape[0])\n",
    "            valuelis *= r.choice([1, np.nan],size = self.df.shape[0] , \n",
    "                                 p = [sample_size/self.df.shape[0], 1-sample_size/self.df.shape[0]])\n",
    "            self.df[['rfid', 'rfid']].assign(trait = valuelis).fillna('NA').astype(str).to_csv(f'{self.path}pvalthresh/randomtrait/{ranid}.txt',  index = False, sep = ' ', header = None)\n",
    "            maxp = 0\n",
    "            for c in self.chrList():\n",
    "                chrom = self.replaceXYMTtonums(c)\n",
    "                filename = f'{self.path}pvalthresh/gwas/{ranid}/chrgwas{c}' \n",
    "                if os.path.exists(f'{filename}.mlma') and skip_already_present: pass\n",
    "                    #printwithlog(f'''skipping gwas for trait: {ranid} and chr {c}''')\n",
    "                else:\n",
    "                    subgrmflag = f'--mlma-subtract-grm {self.path}grm/{c}chrGRM' if c not in ['x','y'] else ''\n",
    "                    bash(f'{self.gcta} --thread-num 1 --pheno {self.path}pvalthresh/randomtrait/{ranid}.txt --bfile {self.path}genotypes/prunedgenotypes \\\n",
    "                                               --grm {self.path}grm/AllchrGRM --autosome-num {self.n_autosome} \\\n",
    "                                               --chr {chrom} {subgrmflag} --mlma \\\n",
    "                                               --out {filename}', \n",
    "                                 print_call = False)#f'GWAS_{chrom}_{ranid}',\n",
    "                if os.path.exists(f'{filename}.mlma'): chrmaxp = np.log10(pd.read_csv(f'{filename}.mlma', sep = '\\t')['p'].min())\n",
    "                else: chrmaxp = 0\n",
    "                if chrmaxp < maxp: maxp = chrmaxp\n",
    "            if remove_after:\n",
    "                bash(f'rm -r {self.path}pvalthresh/gwas/{ranid}')\n",
    "            return maxp\n",
    "        \n",
    "        looppd = pd.DataFrame(range(replicates), columns = ['reps'])\n",
    "        loop   = dd.from_pandas(looppd, npartitions=min(replicates, 200))\n",
    "        # %time get_maxp_1sample(34)\n",
    "        def _gwas_pval(df, rmv_aft):\n",
    "            ret = df['reps'].map(lambda x: get_maxp_1sample(x,  skip_already_present = True, remove_after = rmv_aft)) # skip_already_present = True,remove_after= rmv_aft\n",
    "            if len(df) != len(ret): print(ret)\n",
    "            return ret\n",
    "        _ = loop.map_partitions(lambda x: _gwas_pval(x, rmv_aft = remove_after), meta = pd.Series())\n",
    "        printwithlog(f'running gwas for {replicates} replicates')\n",
    "        future = cline.compute(_)\n",
    "        progress(future,notebook = False,  interval=\"300s\") #, group_by=\"spans\"\n",
    "        wait(future)\n",
    "        out = looppd.assign(maxp = future.result())\n",
    "        for tf in [True, False]:\n",
    "            maxrange = 2000 if tf else len(out)\n",
    "            lis = pd.concat([out.sample(n = x, replace = tf)['maxp'].describe(percentiles=[.1, .05, 0.01, 1e-3, 1e-4]).abs().to_frame().rename({'min': x}, axis = 1) \\\n",
    "                   for x in np.linspace(1, maxrange, 200).round().astype(int)], axis = 1)\n",
    "            lis = lis.T.reset_index(drop = True).rename({'count': 'samplesize'}, axis = 1)\n",
    "            lis = lis.drop(['mean', 'min', 'max'], axis = 1).fillna(0)\n",
    "            melted = lis.melt(id_vars=['samplesize'], value_vars=lis.columns[1:], value_name='pval')\n",
    "            lis.to_csv(f'{self.path}pvalthresh/maxpvaltable{\"with\" if tf else \"without\"}replacement.csv', index = False)\n",
    "            fig = sns.lmplot(x=\"samplesize\", y=\"pval\",\n",
    "                 hue=\"variable\",  data=melted.query('variable != \"std\"'),logx= True,height = 5, aspect=2 )\n",
    "            fig.savefig(f'{self.path}pvalthresh/threshfig{\"with\" if tf else \"without\"}replacement.png')\n",
    "        oo = out['maxp'].describe(percentiles=[.1, .05, 0.01, 1e-3, 1e-4]).to_frame().set_axis(['thresholds'], axis =1).abs()\n",
    "        oo.to_csv(f'{self.path}pvalthresh/PVALTHRESHOLD.csv')\n",
    "        display(oo)\n",
    "        printwithlog(f\"new_thresholds = 5% : {oo.loc['5%','thresholds']} , 10% : {oo.loc['10%','thresholds']}\")\n",
    "        self.threshold = oo.loc['10%','thresholds']\n",
    "        self.threshold05 = oo.loc['5%','thresholds']\n",
    "        return oo   \n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3bbc92f7-a4bf-479d-bda9-99283cd6ee4f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:env-std-py38]",
   "language": "python",
   "name": "conda-env-env-std-py38-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
