{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fbfa856e-8545-4c7a-b2a4-305bbda19007",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“replacing previous import ‘gdata::nobs’ by ‘lavaan::nobs’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘gdata::first’ by ‘data.table::first’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘gdata::last’ by ‘data.table::last’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘gdata::resample’ by ‘R.utils::resample’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘gdata::env’ by ‘R.utils::env’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘data.table::first’ by ‘dplyr::first’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘plyr::arrange’ by ‘dplyr::arrange’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘plyr::count’ by ‘dplyr::count’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘plyr::desc’ by ‘dplyr::desc’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘plyr::summarise’ by ‘dplyr::summarise’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘plyr::id’ by ‘dplyr::id’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘plyr::rename’ by ‘dplyr::rename’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘plyr::summarize’ by ‘dplyr::summarize’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘data.table::last’ by ‘dplyr::last’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘data.table::between’ by ‘dplyr::between’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘gdata::combine’ by ‘dplyr::combine’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘gdata::starts_with’ by ‘dplyr::starts_with’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘plyr::failwith’ by ‘dplyr::failwith’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘plyr::mutate’ by ‘dplyr::mutate’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘Rcpp::.DollarNames’ by ‘utils::.DollarNames’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘Matrix::tail’ by ‘utils::tail’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘gdata::object.size’ by ‘utils::object.size’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘Rcpp::prompt’ by ‘utils::prompt’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘Matrix::head’ by ‘utils::head’ when loading ‘GenomicSEM’”\n",
      "Warning message:\n",
      "“replacing previous import ‘R.utils::timestamp’ by ‘utils::timestamp’ when loading ‘GenomicSEM’”\n",
      "here() starts at /tscc/projects/ps-palmer/brittany/SUD_cross_species/scripts\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(GenomicSEM)\n",
    "library(here)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2bb55981-9932-4d38-9a7a-0358d009f2c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "setwd('/tscc/projects/ps-palmer/brittany/SUD_cross_species/')\n",
    "wd='/tscc/projects/ps-palmer/brittany/SUD_cross_species/'\n",
    "mlma_chr.dir <- (\"/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr\")\n",
    "ref.dir <- (\"scripts/nicsa_munge_ref/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "64645c56-4db3-4dae-b205-659cbdb86a5e",
   "metadata": {},
   "outputs": [],
   "source": [
    "files <- sort(list.files(mlma_chr.dir, full.names = TRUE))\n",
    "trait_tbl=read.table('scripts/nicsa_munge_ref/nicsa_traits.csv',sep=',',header=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "719a6959-937c-4f78-a01a-e676a7f4d8cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "trait_tbl=trait_tbl[order(trait_tbl$file_prefix, decreasing = F),]\n",
    "trait_names=sort(trait_tbl$file_prefix)\n",
    "N=trait_tbl$N\n",
    "se.logit <- c(F, F, F, F, F, F,F, F, F, F, F, F,F, F, F, F, F, F, F)\n",
    "ref <- paste(ref.dir,'nicsa_ref_snplist_notmaf_wchr.tsv',sep='')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "157191e6-5f05-4f0d-9115-509b360508a4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The preparation of 19 summary statistics for use in Genomic SEM began at: 2024-03-18 22:12:41.680685\n",
      "Please note that the files should be in the same order that they were listed for the ldsc function\n",
      "Reading in reference file\n",
      "Applying MAF filer of 0 to the reference file.\n",
      "All files loaded into R!\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_all_days.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_all_days.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 2018 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_all_days.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_all_days.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_all_days.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_all_days.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_all_days.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_all_days.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_all_days.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_all_days.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_first_three_median.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_first_three_median.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 2016 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_first_three_median.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_first_three_median.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_first_three_median.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_first_three_median.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_first_three_median.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_first_three_median.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_first_three_median.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_first_three_median.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_last_three_median.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_last_three_median.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 1950 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_last_three_median.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_last_three_median.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_last_three_median.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_last_three_median.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_last_three_median.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_last_three_median.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_last_three_median.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_last_three_median.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_active_inactive_ratio.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_active_inactive_ratio.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 2002 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_active_inactive_ratio.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_active_inactive_ratio.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_active_inactive_ratio.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_active_inactive_ratio.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_active_inactive_ratio.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_active_inactive_ratio.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_active_inactive_ratio.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_active_inactive_ratio.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_activelick.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_activelick.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 2011 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_activelick.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_activelick.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_activelick.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_activelick.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_activelick.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_activelick.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_activelick.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_activelick.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_infusion.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_infusion.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 2010 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_infusion.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_infusion.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_infusion.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_infusion.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_infusion.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_infusion.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_infusion.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day1_infusion.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_active_inactive_ratio.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_active_inactive_ratio.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 1866 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_active_inactive_ratio.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_active_inactive_ratio.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_active_inactive_ratio.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_active_inactive_ratio.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_active_inactive_ratio.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_active_inactive_ratio.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_active_inactive_ratio.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_active_inactive_ratio.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_activelick.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_activelick.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 1924 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_activelick.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_activelick.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_activelick.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_activelick.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_activelick.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_activelick.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_activelick.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_activelick.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_infusion.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_infusion.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 1925 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_infusion.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_infusion.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_infusion.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_infusion.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_infusion.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_infusion.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_infusion.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_day10_infusion.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_median.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_median.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 2014 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_median.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_median.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_median.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_median.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_median.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_median.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_median.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_median.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_total.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_total.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 2018 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_total.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_total.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_total.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_total.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_total.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_total.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_total.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_activelick_total.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_median.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_median.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 2015 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_median.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_median.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_median.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_median.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_median.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_median.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_median.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_median.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_total.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_total.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 2018 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_total.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_total.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_total.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_total.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_total.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_total.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_total.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_first_three_days_infusion_total.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_median.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_median.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 1989 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_median.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_median.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_median.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_median.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_median.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_median.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_median.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_median.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_total.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_total.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 2018 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_total.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_total.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_total.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_total.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_total.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_total.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_total.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_activelick_total.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_median.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_median.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 1989 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_median.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_median.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_median.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_median.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_median.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_median.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_median.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_median.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_total.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_total.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 2018 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_total.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_total.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_total.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_total.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_total.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_total.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_total.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_last_three_days_infusion_total.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_activelick_10days.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_activelick_10days.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 2018 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_activelick_10days.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_activelick_10days.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_activelick_10days.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_activelick_10days.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_activelick_10days.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_activelick_10days.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_activelick_10days.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_activelick_10days.mlma after QC and merging with the reference file.\n",
      "Preparing summary statistics for file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_infusion_10days.mlma\n",
      "Interpreting the SNP column as the SNP column.\n",
      "Interpreting the A1 column as the A1 column.\n",
      "Interpreting the A2 column as the A2 column.\n",
      "Interpreting the B column as the effect column.\n",
      "Interpreting the P column as the P column.\n",
      "Cannot find N column, try renaming it to N in the summary statistics file for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_infusion_10days.mlma\n",
      "Interpreting the MAF column as the MAF column.\n",
      "Interpreting the SE column as the SE column.\n",
      "Using user provided N of 2018 for /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_infusion_10days.mlma . Please note that this should reflect the sum of effective sample sizes if the linprob argument is being used to back out logistic betas.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_infusion_10days.mlma summary statistics file due to entries that were duplicated for rsID. These are removed as they likely reflect multiallelic variants.\n",
      "Merging file: /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_infusion_10days.mlma with the reference file: scripts/nicsa_munge_ref/nicsa_ref_snplist_notmaf_wchr.tsv\n",
      "5367087 rows present in the full /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_infusion_10days.mlma summary statistics file.\n",
      "0 rows were removed from the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_infusion_10days.mlma summary statistics file as the rsIDs for these SNPs were not present in the reference file.\n",
      "The effect column was determined NOT to be coded as an odds ratio (OR) for the /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_infusion_10days.mlma summary statistics file based on the median of the effect column being close to 0.\n",
      "No INFO column, cannot filter on INFO, which may influence results\n",
      "Performing transformation under the assumption that the effect column is either an odds ratio or logistic beta (please see output above to determine whether it was interpreted as an odds ratio) and the SE column is the SE of the odds ratio (i.e., NOT on the logistic scale) for:/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_infusion_10days.mlma\n",
      "5367087 SNPs are left in the summary statistics file /tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_total_infusion_10days.mlma after QC and merging with the reference file.\n",
      "After merging across all summary statistics using listwise deletion, performing QC, and merging with the reference file, there are 5367087 SNPs left in the final multivariate summary statistics file\n",
      "Sumstats finished running at 2024-03-18 22:24:27.630739\n",
      "Running sumstats for all files took 11 minutes and 45.9500546455383 seconds\n",
      "Please check the log file regressedlr_nicsa_active_inactive_ratio_all_days_regressedlr_nicsa_active_inactive_ratio_first_three_sumstats.log to ensure that all columns were interpreted correctly and no warnings were issued for any of the summary statistics files.\n"
     ]
    }
   ],
   "source": [
    "SUMSTATSoutput <- sumstats(files=files, ref = ref, N=N,\n",
    "       trait.names = trait_names, linprob=se.logit, \n",
    "                           se.logit = se.logit, maf.filter = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "8c21c6d0-0844-493a-85a1-68a63e8ff4d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "save(SUMSTATSoutput, file=\"SUMSTATSoutput.RData\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "379570f8-c852-4b66-bec5-4cb46783e590",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_tbl=read.table(ref,sep='\\t',header=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "785035c2-2e9a-47c3-a8d1-e61555a5eaa6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'SNP'</li><li>'Chr'</li><li>'bp'</li><li>'MAF'</li><li>'A1'</li><li>'A2'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SNP'\n",
       "\\item 'Chr'\n",
       "\\item 'bp'\n",
       "\\item 'MAF'\n",
       "\\item 'A1'\n",
       "\\item 'A2'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SNP'\n",
       "2. 'Chr'\n",
       "3. 'bp'\n",
       "4. 'MAF'\n",
       "5. 'A1'\n",
       "6. 'A2'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"SNP\" \"Chr\" \"bp\"  \"MAF\" \"A1\"  \"A2\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(ref_tbl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "6127ea6e-faea-4504-b98c-d85b330b27af",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_tbl=read.table('/tscc/nfs/home/bsleger/bsl/SUD_cross_species/nicsa_gwas/mlma/mlma_concat/mlma_concat_wchr/regressedlr_nicsa_active_inactive_ratio_all_days.mlma',sep='\\t',header=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "02c14c5f-8f5e-4f42-b131-4d8b547601b0",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in eval(expr, envir, enclos): object 'SUMSTATSoutput' not found\n",
     "output_type": "error",
     "traceback": [
      "Error in eval(expr, envir, enclos): object 'SUMSTATSoutput' not found\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "SUMSTATSoutput"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "907b8d5c-fa4b-4c9c-8c7b-b46fb1724eb9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<table style=\"width: 100%;\"><tr><td>sumstats {GenomicSEM}</td><td style=\"text-align: right;\">R Documentation</td></tr></table>\n",
       "\n",
       "<h2 id='sumstats'>Allign summary statistics from univariate GWAS for a GWAS in GenomicSEM</h2>\n",
       "\n",
       "<h3>Description</h3>\n",
       "\n",
       "<p>Function to process GWAS summary statistics files and prepare them for a GWAS in genomicSEM\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Usage</h3>\n",
       "\n",
       "<pre><code class='language-R'>sumstats(files,ref,trait.names=NULL,se.logit,OLS=NULL,linprob=NULL,N=NULL,betas=NULL,info.filter = .6,maf.filter=0.01,\n",
       "         keep.indel=FALSE,parallel=FALSE,cores=NULL,ambig=FALSE,direct.filter=FALSE, ...)\n",
       "</code></pre>\n",
       "\n",
       "\n",
       "<h3>Arguments</h3>\n",
       "\n",
       "<table>\n",
       "<tr><td><code id=\"sumstats_:_files\">files</code></td>\n",
       "<td>\n",
       "<p>a vector of file names, files must be located in the working directory, or a path must be provided.</p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_ref\">ref</code></td>\n",
       "<td>\n",
       "<p>A reference file of SNPs to keep in your GWAS, one based on 1000 genomes phase 3 is provided. </p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_trait.names\">trait.names</code></td>\n",
       "<td>\n",
       "<p> a vector of trait names which will be used as names for the munged files</p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_se.logit\">se.logit</code></td>\n",
       "<td>\n",
       "<p> a logical vector indicating whether the standard errors in each set of summary statistics is on the logit scale</p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_OLS\">OLS</code></td>\n",
       "<td>\n",
       "<p> a logical vector indicating whether the GWAS was for a continuous trait and used OLS (or a LMM)</p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_linprob\">linprob</code></td>\n",
       "<td>\n",
       "<p>a logical vector indicating whether the GWAS is a binary outcome with only Z-statistics or was analyzed using a linear probability model i.e. a dichotomous trait using OLS (or a LMM)</p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_N\">N</code></td>\n",
       "<td>\n",
       "<p>A vector of total sample sizes for continuous traits and the sum of effective sample sizes for binary traits</p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_betas\">betas</code></td>\n",
       "<td>\n",
       "<p>A vector of column names of betas for continuous traits that are known to have been standardized prior to running the GWAS</p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_N\">N</code></td>\n",
       "<td>\n",
       "<p>A vector of sample size</p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_info.filter\">info.filter</code></td>\n",
       "<td>\n",
       "<p> Numeric value which is used as a lower bound for imputation quality (INFO) </p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_maf.filter\">maf.filter</code></td>\n",
       "<td>\n",
       "<p> Numeric value used as a lower bound for minor allele frequency</p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_keep.indel\">keep.indel</code></td>\n",
       "<td>\n",
       "<p>Indicates whether insertion-deletion mutations (indels) should be included in your summary statistics. The default is FALSE.</p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_parallel\">parallel</code></td>\n",
       "<td>\n",
       "<p>Indicates whether sumstats should process the summary statistics files in parallel or serial fashion. Default is TRUE, indicating that it will run in parallel.</p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_cores\">cores</code></td>\n",
       "<td>\n",
       "<p>Indicates how many cores to use when running in parallel. The default is NULL, in which case sumstats will use 1 less than the total number of cores available in the local environment.</p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_ambig\">ambig</code></td>\n",
       "<td>\n",
       "<p>Indicates whether strand ambiguous SNPs should be removed from output. </p>\n",
       "</td></tr>\n",
       "<tr><td><code id=\"sumstats_:_direct.filter\">direct.filter</code></td>\n",
       "<td>\n",
       "<p>Indicates whether SNPs that have missing information for more than half of contributing cohorts, as indicated by missing information in the direction column, should be removed.</p>\n",
       "</td></tr>        \n",
       "</table>\n",
       "\n",
       "\n",
       "<h3>Value</h3>\n",
       "\n",
       "<p>The function ensures the SNPs in each file are aligned to the same reference allele, it attempts to filter strand issues, it retains SNPs present in the reference file. The function can deal with GWAS of continous traits, dichotomous traits using logistic regression and even dichotomous traits using (misspecified) OLS regression or a mixed model. \n",
       "The function returns .log files that should be inspected to ensure that all column names were appropriately interpreted. \n",
       "</p>\n",
       "\n",
       "<hr /><div style=\"text-align: center;\">[Package <em>GenomicSEM</em> version 0.0.5 ]</div>\n",
       "</div>"
      ],
      "text/latex": [
       "\\inputencoding{utf8}\n",
       "\\HeaderA{sumstats}{Allign summary statistics from univariate GWAS for a GWAS in GenomicSEM}{sumstats}\n",
       "%\n",
       "\\begin{Description}\n",
       "Function to process GWAS summary statistics files and prepare them for a GWAS in genomicSEM\n",
       "\\end{Description}\n",
       "%\n",
       "\\begin{Usage}\n",
       "\\begin{verbatim}\n",
       "sumstats(files,ref,trait.names=NULL,se.logit,OLS=NULL,linprob=NULL,N=NULL,betas=NULL,info.filter = .6,maf.filter=0.01,\n",
       "         keep.indel=FALSE,parallel=FALSE,cores=NULL,ambig=FALSE,direct.filter=FALSE, ...)\n",
       "\\end{verbatim}\n",
       "\\end{Usage}\n",
       "%\n",
       "\\begin{Arguments}\n",
       "\\begin{ldescription}\n",
       "\\item[\\code{files}] a vector of file names, files must be located in the working directory, or a path must be provided.\n",
       "\\item[\\code{ref}] A reference file of SNPs to keep in your GWAS, one based on 1000 genomes phase 3 is provided. \n",
       "\\item[\\code{trait.names}]  a vector of trait names which will be used as names for the munged files\n",
       "\\item[\\code{se.logit}]  a logical vector indicating whether the standard errors in each set of summary statistics is on the logit scale\n",
       "\\item[\\code{OLS}]  a logical vector indicating whether the GWAS was for a continuous trait and used OLS (or a LMM)\n",
       "\\item[\\code{linprob}] a logical vector indicating whether the GWAS is a binary outcome with only Z-statistics or was analyzed using a linear probability model i.e. a dichotomous trait using OLS (or a LMM)\n",
       "\\item[\\code{N}] A vector of total sample sizes for continuous traits and the sum of effective sample sizes for binary traits\n",
       "\\item[\\code{betas}] A vector of column names of betas for continuous traits that are known to have been standardized prior to running the GWAS\n",
       "\\item[\\code{N}] A vector of sample size\n",
       "\\item[\\code{info.filter}]  Numeric value which is used as a lower bound for imputation quality (INFO) \n",
       "\\item[\\code{maf.filter}]  Numeric value used as a lower bound for minor allele frequency\n",
       "\\item[\\code{keep.indel}] Indicates whether insertion-deletion mutations (indels) should be included in your summary statistics. The default is FALSE.\n",
       "\\item[\\code{parallel}] Indicates whether sumstats should process the summary statistics files in parallel or serial fashion. Default is TRUE, indicating that it will run in parallel.\n",
       "\\item[\\code{cores}] Indicates how many cores to use when running in parallel. The default is NULL, in which case sumstats will use 1 less than the total number of cores available in the local environment.\n",
       "\\item[\\code{ambig}] Indicates whether strand ambiguous SNPs should be removed from output. \n",
       "\\item[\\code{direct.filter}] Indicates whether SNPs that have missing information for more than half of contributing cohorts, as indicated by missing information in the direction column, should be removed.       \n",
       "\n",
       "\\end{ldescription}\n",
       "\\end{Arguments}\n",
       "%\n",
       "\\begin{Value}\n",
       "The function ensures the SNPs in each file are aligned to the same reference allele, it attempts to filter strand issues, it retains SNPs present in the reference file. The function can deal with GWAS of continous traits, dichotomous traits using logistic regression and even dichotomous traits using (misspecified) OLS regression or a mixed model. \n",
       "The function returns .log files that should be inspected to ensure that all column names were appropriately interpreted. \n",
       "\n",
       "\\end{Value}"
      ],
      "text/plain": [
       "sumstats              package:GenomicSEM               R Documentation\n",
       "\n",
       "_\bA_\bl_\bl_\bi_\bg_\bn _\bs_\bu_\bm_\bm_\ba_\br_\by _\bs_\bt_\ba_\bt_\bi_\bs_\bt_\bi_\bc_\bs _\bf_\br_\bo_\bm _\bu_\bn_\bi_\bv_\ba_\br_\bi_\ba_\bt_\be _\bG_\bW_\bA_\bS _\bf_\bo_\br _\ba _\bG_\bW_\bA_\bS _\bi_\bn _\bG_\be_\bn_\bo_\bm_\bi_\bc_\bS_\bE_\bM\n",
       "\n",
       "_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n",
       "\n",
       "     Function to process GWAS summary statistics files and prepare them\n",
       "     for a GWAS in genomicSEM\n",
       "\n",
       "_\bU_\bs_\ba_\bg_\be:\n",
       "\n",
       "     sumstats(files,ref,trait.names=NULL,se.logit,OLS=NULL,linprob=NULL,N=NULL,betas=NULL,info.filter = .6,maf.filter=0.01,\n",
       "              keep.indel=FALSE,parallel=FALSE,cores=NULL,ambig=FALSE,direct.filter=FALSE, ...)\n",
       "     \n",
       "_\bA_\br_\bg_\bu_\bm_\be_\bn_\bt_\bs:\n",
       "\n",
       "   files: a vector of file names, files must be located in the working\n",
       "          directory, or a path must be provided.\n",
       "\n",
       "     ref: A reference file of SNPs to keep in your GWAS, one based on\n",
       "          1000 genomes phase 3 is provided.\n",
       "\n",
       "trait.names: a vector of trait names which will be used as names for\n",
       "          the munged files\n",
       "\n",
       "se.logit: a logical vector indicating whether the standard errors in\n",
       "          each set of summary statistics is on the logit scale\n",
       "\n",
       "     OLS: a logical vector indicating whether the GWAS was for a\n",
       "          continuous trait and used OLS (or a LMM)\n",
       "\n",
       " linprob: a logical vector indicating whether the GWAS is a binary\n",
       "          outcome with only Z-statistics or was analyzed using a linear\n",
       "          probability model i.e. a dichotomous trait using OLS (or a\n",
       "          LMM)\n",
       "\n",
       "       N: A vector of total sample sizes for continuous traits and the\n",
       "          sum of effective sample sizes for binary traits\n",
       "\n",
       "   betas: A vector of column names of betas for continuous traits that\n",
       "          are known to have been standardized prior to running the GWAS\n",
       "\n",
       "       N: A vector of sample size\n",
       "\n",
       "info.filter: Numeric value which is used as a lower bound for\n",
       "          imputation quality (INFO)\n",
       "\n",
       "maf.filter: Numeric value used as a lower bound for minor allele\n",
       "          frequency\n",
       "\n",
       "keep.indel: Indicates whether insertion-deletion mutations (indels)\n",
       "          should be included in your summary statistics. The default is\n",
       "          FALSE.\n",
       "\n",
       "parallel: Indicates whether sumstats should process the summary\n",
       "          statistics files in parallel or serial fashion. Default is\n",
       "          TRUE, indicating that it will run in parallel.\n",
       "\n",
       "   cores: Indicates how many cores to use when running in parallel. The\n",
       "          default is NULL, in which case sumstats will use 1 less than\n",
       "          the total number of cores available in the local environment.\n",
       "\n",
       "   ambig: Indicates whether strand ambiguous SNPs should be removed\n",
       "          from output.\n",
       "\n",
       "direct.filter: Indicates whether SNPs that have missing information for\n",
       "          more than half of contributing cohorts, as indicated by\n",
       "          missing information in the direction column, should be\n",
       "          removed.\n",
       "\n",
       "_\bV_\ba_\bl_\bu_\be:\n",
       "\n",
       "     The function ensures the SNPs in each file are aligned to the same\n",
       "     reference allele, it attempts to filter strand issues, it retains\n",
       "     SNPs present in the reference file. The function can deal with\n",
       "     GWAS of continous traits, dichotomous traits using logistic\n",
       "     regression and even dichotomous traits using (misspecified) OLS\n",
       "     regression or a mixed model.  The function returns .log files that\n",
       "     should be inspected to ensure that all column names were\n",
       "     appropriately interpreted.\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "help(sumstats)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:lzenv]",
   "language": "R",
   "name": "conda-env-lzenv-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
