{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "bc2a1de5-cf96-4c99-9fa7-d19d1c35e8bf",
   "metadata": {},
   "source": [
    "purpose: run MAGMA on human control traits (taken from the ext1.0 paper's traits that were least and most significantly associated with externalizing)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ce4474f3-45db-429b-b298-d0c77efe6525",
   "metadata": {},
   "source": [
    "# set-up"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "10b4edc0-319b-4190-9fed-05d16c1be78e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import glob\n",
    "import statsmodels.stats.multitest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b9499dec-d876-4d0a-8372-73ff699b2b55",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('/tscc/projects/ps-palmer/brittany/SUD_cross_species/gwas_ctrl_hm')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2c6dc167-5d12-4382-b057-cdbe8e22e5a2",
   "metadata": {},
   "source": [
    "# import datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "78fd11b6-4fa8-4581-b746-4696de201a5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "meta_data=pd.read_csv('ext1_st8_corr_sources_downloaded.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3c1146de-befc-442a-a851-3e5024baa488",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Trait 1</th>\n",
       "      <th>Trait 2</th>\n",
       "      <th>N trait 2</th>\n",
       "      <th>rG</th>\n",
       "      <th>abs(rG)</th>\n",
       "      <th>SE</th>\n",
       "      <th>Z</th>\n",
       "      <th>P value</th>\n",
       "      <th>Source</th>\n",
       "      <th>Domain</th>\n",
       "      <th>Subdomain</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>N</th>\n",
       "      <th>Prepared?</th>\n",
       "      <th>Download</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Externalizing</td>\n",
       "      <td>Townsend deprivation index*</td>\n",
       "      <td>112,151</td>\n",
       "      <td>0.707</td>\n",
       "      <td>0.707</td>\n",
       "      <td>0.047</td>\n",
       "      <td>15.167</td>\n",
       "      <td>5.820000e-52</td>\n",
       "      <td>CCACE</td>\n",
       "      <td>Socioeconomic</td>\n",
       "      <td>Economic &amp; Environmental</td>\n",
       "      <td>Neighborhood deprivation (Townsend Index)</td>\n",
       "      <td>112151</td>\n",
       "      <td>YES</td>\n",
       "      <td>http://www.psy.ed.ac.uk/ccace/downloads/Hill_C...</td>\n",
       "      <td>townsend</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Externalizing</td>\n",
       "      <td>Antisocial behavior</td>\n",
       "      <td>16,400</td>\n",
       "      <td>0.647</td>\n",
       "      <td>0.647</td>\n",
       "      <td>0.166</td>\n",
       "      <td>3.906</td>\n",
       "      <td>9.400000e-05</td>\n",
       "      <td>CTG Lab</td>\n",
       "      <td>Psychopathology</td>\n",
       "      <td>Psychiatric Symptoms</td>\n",
       "      <td>Antisocial behavior</td>\n",
       "      <td>16400</td>\n",
       "      <td>YES</td>\n",
       "      <td>http://broadabc.ctglab.nl/documents/p12/BroadA...</td>\n",
       "      <td>antisoc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Externalizing</td>\n",
       "      <td>Age of smoking initiation*</td>\n",
       "      <td>262,990</td>\n",
       "      <td>-0.646</td>\n",
       "      <td>0.646</td>\n",
       "      <td>0.026</td>\n",
       "      <td>-24.708</td>\n",
       "      <td>8.760000e-135</td>\n",
       "      <td>GSCAN</td>\n",
       "      <td>Psychopathology</td>\n",
       "      <td>Substance Use</td>\n",
       "      <td>Age of smoking initiation</td>\n",
       "      <td>262990</td>\n",
       "      <td>YES</td>\n",
       "      <td>https://conservancy.umn.edu/handle/11299/201564</td>\n",
       "      <td>age_smkinit</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Externalizing</td>\n",
       "      <td>Friendships satisfaction*</td>\n",
       "      <td>361,194</td>\n",
       "      <td>-0.001</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.032</td>\n",
       "      <td>-0.039</td>\n",
       "      <td>9.680000e-01</td>\n",
       "      <td>GWAS Atlas</td>\n",
       "      <td>Socioeconomic</td>\n",
       "      <td>Social</td>\n",
       "      <td>Friendships satisfaction</td>\n",
       "      <td>127735</td>\n",
       "      <td>YES</td>\n",
       "      <td>https://atlas.ctglab.nl/ukb2_sumstats/f.4570.0...</td>\n",
       "      <td>friend_sat</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Externalizing</td>\n",
       "      <td>Infant birth weight*</td>\n",
       "      <td>361,194</td>\n",
       "      <td>-0.012</td>\n",
       "      <td>0.012</td>\n",
       "      <td>0.020</td>\n",
       "      <td>-0.609</td>\n",
       "      <td>5.420000e-01</td>\n",
       "      <td>GWAS Atlas</td>\n",
       "      <td>Health &amp; Medicine</td>\n",
       "      <td>Anthropomorphic</td>\n",
       "      <td>Infant birth weight</td>\n",
       "      <td>219088</td>\n",
       "      <td>YES</td>\n",
       "      <td>https://atlas.ctglab.nl/ukb2_sumstats/f.20022....</td>\n",
       "      <td>infant_bw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Externalizing</td>\n",
       "      <td>Heart rate*</td>\n",
       "      <td>361,194</td>\n",
       "      <td>-0.026</td>\n",
       "      <td>0.026</td>\n",
       "      <td>0.018</td>\n",
       "      <td>-1.426</td>\n",
       "      <td>1.540000e-01</td>\n",
       "      <td>GWAS Atlas</td>\n",
       "      <td>Health &amp; Medicine</td>\n",
       "      <td>Cardiometabolic</td>\n",
       "      <td>Heart rate</td>\n",
       "      <td>361411</td>\n",
       "      <td>YES</td>\n",
       "      <td>https://atlas.ctglab.nl/ukb2_sumstats/f.102.0....</td>\n",
       "      <td>hr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Externalizing</td>\n",
       "      <td>Age at first facial hair (male)*</td>\n",
       "      <td>167,020</td>\n",
       "      <td>0.033</td>\n",
       "      <td>0.033</td>\n",
       "      <td>0.024</td>\n",
       "      <td>1.420</td>\n",
       "      <td>1.560000e-01</td>\n",
       "      <td>GWAS Atlas</td>\n",
       "      <td>Demography</td>\n",
       "      <td>Life History &amp; Reproduction</td>\n",
       "      <td>Relative age of first facial hair (male)</td>\n",
       "      <td>171805</td>\n",
       "      <td>YES</td>\n",
       "      <td>https://atlas.ctglab.nl/ukb2_sumstats/f.2375.0...</td>\n",
       "      <td>facial_hair</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Externalizing</td>\n",
       "      <td>Maternal smoking around birth*</td>\n",
       "      <td>361,194</td>\n",
       "      <td>0.717</td>\n",
       "      <td>0.717</td>\n",
       "      <td>0.021</td>\n",
       "      <td>34.403</td>\n",
       "      <td>2.240000e-259</td>\n",
       "      <td>GWAS Atlas</td>\n",
       "      <td>Socioeconomic</td>\n",
       "      <td>Economic &amp; Environmental</td>\n",
       "      <td>Maternal smoking around birth</td>\n",
       "      <td>331862</td>\n",
       "      <td>YES</td>\n",
       "      <td>https://atlas.ctglab.nl/ukb2_sumstats/f.1787.0...</td>\n",
       "      <td>maternal_smok</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Externalizing</td>\n",
       "      <td>Age at menarche*</td>\n",
       "      <td>194,174</td>\n",
       "      <td>-0.010</td>\n",
       "      <td>0.010</td>\n",
       "      <td>0.021</td>\n",
       "      <td>-0.461</td>\n",
       "      <td>6.450000e-01</td>\n",
       "      <td>ReproGen</td>\n",
       "      <td>Demography</td>\n",
       "      <td>Life History &amp; Reproduction</td>\n",
       "      <td>Age at menarche</td>\n",
       "      <td>252514</td>\n",
       "      <td>YES</td>\n",
       "      <td>http://www.reprogen.org/Menarche_1KG_NatGen201...</td>\n",
       "      <td>age_menarche</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Externalizing</td>\n",
       "      <td>LDL</td>\n",
       "      <td>173,081</td>\n",
       "      <td>0.025</td>\n",
       "      <td>0.025</td>\n",
       "      <td>0.032</td>\n",
       "      <td>0.768</td>\n",
       "      <td>4.430000e-01</td>\n",
       "      <td>GLGC</td>\n",
       "      <td>Health &amp; Medicine</td>\n",
       "      <td>Cardiometabolic</td>\n",
       "      <td>LDL Cholesterol</td>\n",
       "      <td>188577</td>\n",
       "      <td>YES</td>\n",
       "      <td>http://csg.sph.umich.edu/abecasis/public/lipid...</td>\n",
       "      <td>LDL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Externalizing</td>\n",
       "      <td>Neuroticism**</td>\n",
       "      <td>59,206</td>\n",
       "      <td>0.022</td>\n",
       "      <td>0.022</td>\n",
       "      <td>0.050</td>\n",
       "      <td>0.438</td>\n",
       "      <td>6.620000e-01</td>\n",
       "      <td>GPC</td>\n",
       "      <td>Psychopathology</td>\n",
       "      <td>Personality - Big 5</td>\n",
       "      <td>Neuroticism</td>\n",
       "      <td>63661</td>\n",
       "      <td>YES</td>\n",
       "      <td>https://www.dropbox.com/s/ql1en6s0kramxne/GPC-...</td>\n",
       "      <td>neurot</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Trait 1                           Trait 2 N trait 2     rG  abs(rG)  \\\n",
       "0   Externalizing       Townsend deprivation index*   112,151  0.707    0.707   \n",
       "1   Externalizing               Antisocial behavior    16,400  0.647    0.647   \n",
       "2   Externalizing        Age of smoking initiation*   262,990 -0.646    0.646   \n",
       "3   Externalizing         Friendships satisfaction*   361,194 -0.001    0.001   \n",
       "4   Externalizing              Infant birth weight*   361,194 -0.012    0.012   \n",
       "5   Externalizing                       Heart rate*   361,194 -0.026    0.026   \n",
       "6   Externalizing  Age at first facial hair (male)*   167,020  0.033    0.033   \n",
       "7   Externalizing    Maternal smoking around birth*   361,194  0.717    0.717   \n",
       "8   Externalizing                  Age at menarche*   194,174 -0.010    0.010   \n",
       "9   Externalizing                               LDL   173,081  0.025    0.025   \n",
       "10  Externalizing                     Neuroticism**    59,206  0.022    0.022   \n",
       "\n",
       "       SE       Z        P value      Source             Domain  \\\n",
       "0   0.047  15.167   5.820000e-52       CCACE      Socioeconomic   \n",
       "1   0.166   3.906   9.400000e-05     CTG Lab    Psychopathology   \n",
       "2   0.026 -24.708  8.760000e-135       GSCAN    Psychopathology   \n",
       "3   0.032  -0.039   9.680000e-01  GWAS Atlas      Socioeconomic   \n",
       "4   0.020  -0.609   5.420000e-01  GWAS Atlas  Health & Medicine   \n",
       "5   0.018  -1.426   1.540000e-01  GWAS Atlas  Health & Medicine   \n",
       "6   0.024   1.420   1.560000e-01  GWAS Atlas         Demography   \n",
       "7   0.021  34.403  2.240000e-259  GWAS Atlas      Socioeconomic   \n",
       "8   0.021  -0.461   6.450000e-01    ReproGen         Demography   \n",
       "9   0.032   0.768   4.430000e-01        GLGC  Health & Medicine   \n",
       "10  0.050   0.438   6.620000e-01         GPC    Psychopathology   \n",
       "\n",
       "                      Subdomain                                  Phenotype  \\\n",
       "0      Economic & Environmental  Neighborhood deprivation (Townsend Index)   \n",
       "1          Psychiatric Symptoms                        Antisocial behavior   \n",
       "2                 Substance Use                  Age of smoking initiation   \n",
       "3                        Social                   Friendships satisfaction   \n",
       "4               Anthropomorphic                        Infant birth weight   \n",
       "5               Cardiometabolic                                 Heart rate   \n",
       "6   Life History & Reproduction   Relative age of first facial hair (male)   \n",
       "7      Economic & Environmental              Maternal smoking around birth   \n",
       "8   Life History & Reproduction                            Age at menarche   \n",
       "9               Cardiometabolic                            LDL Cholesterol   \n",
       "10          Personality - Big 5                                Neuroticism   \n",
       "\n",
       "         N Prepared?                                           Download  \\\n",
       "0   112151       YES  http://www.psy.ed.ac.uk/ccace/downloads/Hill_C...   \n",
       "1    16400       YES  http://broadabc.ctglab.nl/documents/p12/BroadA...   \n",
       "2   262990       YES    https://conservancy.umn.edu/handle/11299/201564   \n",
       "3   127735       YES  https://atlas.ctglab.nl/ukb2_sumstats/f.4570.0...   \n",
       "4   219088       YES  https://atlas.ctglab.nl/ukb2_sumstats/f.20022....   \n",
       "5   361411       YES  https://atlas.ctglab.nl/ukb2_sumstats/f.102.0....   \n",
       "6   171805       YES  https://atlas.ctglab.nl/ukb2_sumstats/f.2375.0...   \n",
       "7   331862       YES  https://atlas.ctglab.nl/ukb2_sumstats/f.1787.0...   \n",
       "8   252514       YES  http://www.reprogen.org/Menarche_1KG_NatGen201...   \n",
       "9   188577       YES  http://csg.sph.umich.edu/abecasis/public/lipid...   \n",
       "10   63661       YES  https://www.dropbox.com/s/ql1en6s0kramxne/GPC-...   \n",
       "\n",
       "            label  \n",
       "0        townsend  \n",
       "1         antisoc  \n",
       "2     age_smkinit  \n",
       "3      friend_sat  \n",
       "4       infant_bw  \n",
       "5              hr  \n",
       "6     facial_hair  \n",
       "7   maternal_smok  \n",
       "8    age_menarche  \n",
       "9             LDL  \n",
       "10         neurot  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0ef6c6ee-fc7c-4cc5-bd17-be18a585c54d",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_dict={'facial_hair':pd.read_csv('age_facial_hair_male.txt.gz',compression='gzip',sep='\\t'),\n",
    "    'age_smkinit':pd.read_csv('age_smkinit.txt.gz',compression='gzip',sep='\\t'),\n",
    "    'antisoc':pd.read_csv('antisoc_behav.tbl',sep='\\t',low_memory=False),\n",
    "    'friend_sat':pd.read_csv('friendship_satisfaction.txt.gz',compression='gzip',sep='\\t'),\n",
    "    'hr':pd.read_csv('heart_rate.txt',sep='\\t'),\n",
    "    'infant_bw':pd.read_csv('infant_birth_weight.txt.gz',compression='gzip',sep='\\t'),\n",
    "    'LDL':pd.read_csv('jointGwasMc_LDL.txt.gz',compression='gzip',sep='\\t'),\n",
    "    'maternal_smok':pd.read_csv('maternal_smok.txt',sep='\\t'),\n",
    "    'townsend':pd.read_csv('townsend_Hill_CB_2016/Hill2016_UKB_Income_summary_results_21112016.txt',sep=' '),\n",
    "    'age_menarche':pd.read_csv('age_menarche/Menarche_1KG_NatGen2017_WebsiteUpload.txt',delim_whitespace=True),\n",
    "    'neurot':pd.read_csv('neuroticism/GPC-2.NEUROTICISM.full.txt',sep='\\t',header=None)   \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f486202a-86b6-468b-96ef-f6f0c388c0ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_dict['antisoc']['CHR']=file_dict['antisoc']['MarkerName'].apply(lambda s: s.split(':')[0])\n",
    "file_dict['antisoc']['BP']=file_dict['antisoc']['MarkerName'].apply(lambda s: s.split(':')[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "13055434-1bbc-483b-af4e-3136b50afddd",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_dict['neurot'].columns=['SNPID','CHR','BP','A1','A2','BETA','SE','PVALUE','NCOH','MAF']\n",
    "#from the neuroticism read me"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2a97399d-524e-4cae-8218-e61fd9e1c69b",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_dict['age_menarche']['CHR']=file_dict['age_menarche']['Markername'].apply(lambda s: s.split(':')[0][3:])\n",
    "file_dict['age_menarche']['BP']=file_dict['age_menarche']['Markername'].apply(lambda s: s.split(':')[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e82a442c-99d5-4d51-a8dc-755d92e56a4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_dict['LDL']['CHR']=file_dict['LDL']['SNP_hg19'].apply(lambda s: s.split(':')[0][3:])\n",
    "file_dict['LDL']['BP']=file_dict['LDL']['SNP_hg19'].apply(lambda s: s.split(':')[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "83b65a76-48d3-453d-bcca-b1c8a2f11595",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "facial_hair\n",
      "Index(['SNP', 'CHR', 'BP', 'A1', 'TEST', 'NMISS', 'BETA', 'SE', 'L95', 'U95',\n",
      "       'STAT', 'P', 'A2', 'MAF', 'NCHROBS', 'SNPID_UKB', 'A1_UKB', 'A2_UKB',\n",
      "       'INFO_UKB', 'MAF_UKB'],\n",
      "      dtype='object')\n",
      "age_smkinit\n",
      "Index(['CHROM', 'POS', 'RSID', 'REF', 'ALT', 'AF', 'STAT', 'PVALUE', 'BETA',\n",
      "       'SE', 'N', 'EFFECTIVE_N', 'Number_of_Studies', 'ANNO', 'ANNOFULL'],\n",
      "      dtype='object')\n",
      "antisoc\n",
      "Index(['MarkerName', 'Allele1', 'Allele2', 'Weight', 'Zscore', 'P-value',\n",
      "       'Direction', 'CHR', 'BP'],\n",
      "      dtype='object')\n",
      "friend_sat\n",
      "Index(['SNP', 'CHR', 'BP', 'A1', 'TEST', 'NMISS', 'BETA', 'SE', 'L95', 'U95',\n",
      "       'STAT', 'P', 'A2', 'MAF', 'NCHROBS', 'SNPID_UKB', 'A1_UKB', 'A2_UKB',\n",
      "       'INFO_UKB', 'MAF_UKB'],\n",
      "      dtype='object')\n",
      "hr\n",
      "Index(['SNP', 'CHR', 'BP', 'A1', 'TEST', 'NMISS', 'BETA', 'SE', 'L95', 'U95',\n",
      "       'STAT', 'P', 'A2', 'MAF', 'NCHROBS', 'SNPID_UKB', 'A1_UKB', 'A2_UKB',\n",
      "       'INFO_UKB', 'MAF_UKB'],\n",
      "      dtype='object')\n",
      "infant_bw\n",
      "Index(['SNP', 'CHR', 'BP', 'A1', 'TEST', 'NMISS', 'BETA', 'SE', 'L95', 'U95',\n",
      "       'STAT', 'P', 'A2', 'MAF', 'NCHROBS', 'SNPID_UKB', 'A1_UKB', 'A2_UKB',\n",
      "       'INFO_UKB', 'MAF_UKB'],\n",
      "      dtype='object')\n",
      "LDL\n",
      "Index(['SNP_hg18', 'SNP_hg19', 'rsid', 'A1', 'A2', 'beta', 'se', 'N',\n",
      "       'P-value', 'Freq.A1.1000G.EUR', 'CHR', 'BP'],\n",
      "      dtype='object')\n",
      "maternal_smok\n",
      "Index(['SNP', 'CHR', 'BP', 'A1', 'TEST', 'NMISS', 'OR', 'SE', 'L95', 'U95',\n",
      "       'STAT', 'P', 'A2', 'MAF', 'NCHROBS', 'SNPID_UKB', 'A1_UKB', 'A2_UKB',\n",
      "       'INFO_UKB', 'MAF_UKB'],\n",
      "      dtype='object')\n",
      "townsend\n",
      "Index(['Chromosome', 'Position', 'Markername', 'Effect_allele', 'Other_allele',\n",
      "       'P-value', 'Beta'],\n",
      "      dtype='object')\n",
      "age_menarche\n",
      "Index(['Markername', 'Allele1', 'Allele2', 'Effect', 'Pvalue', 'Minor_Allele',\n",
      "       'CHR', 'BP'],\n",
      "      dtype='object')\n",
      "neurot\n",
      "Index(['SNPID', 'CHR', 'BP', 'A1', 'A2', 'BETA', 'SE', 'PVALUE', 'NCOH',\n",
      "       'MAF'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "for f in file_dict.keys():\n",
    "    print(f)\n",
    "    print(file_dict[f].columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "2d3abe92-fd8b-4dab-9624-2bc644c926b6",
   "metadata": {},
   "outputs": [],
   "source": [
    "colname_dict={\n",
    "    'facial_hair':['SNP','CHR','BP','P','A1','A2'],\n",
    "    'age_smkinit':['RSID','CHROM','POS','PVALUE'],\n",
    "    'antisoc':['MarkerName','CHR','BP','P-value','Allele1','Allele2'],\n",
    "    'friend_sat':['SNP','CHR','BP','P','A1','A2'], \n",
    "    'hr':['SNP','CHR','BP','P','A1','A2'],\n",
    "    'infant_bw':['SNP','CHR','BP','P','A1','A2'],\n",
    "    'LDL':['rsid','CHR','BP','P-value'], \n",
    "    'maternal_smok':['SNP','CHR','BP','P','A1','A2'],\n",
    "    'townsend':['Markername','Chromosome','Position','P-value'],\n",
    "    'age_menarche':['Markername','CHR','BP','Pvalue','Allele1','Allele2'], \n",
    "    'neurot':['SNPID','CHR','BP','PVALUE']\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "17b9ee80-9d42-4d16-9fff-4061cd4af850",
   "metadata": {},
   "outputs": [],
   "source": [
    "def ex_magma(snp,colname,N,prefix,dir,use_RSID=False):\n",
    "    #purpose- format pvalue and pos files- specifics for mlma\n",
    "    if use_RSID==False:\n",
    "        pos=snp[[colname[0],colname[1],colname[2]]]\n",
    "    else:\n",
    "        pos=snp[['RSID',colname[1],colname[2]]]\n",
    "    pos.columns=['SNP','CHR','POS']\n",
    "    pos.to_csv(dir+prefix+'_pos.tsv',sep='\\t',index=False)\n",
    "    print(pos.head())\n",
    "    print('pos file written to '+dir+prefix+'_pos.tsv')\n",
    "    if use_RSID==False:\n",
    "        pval=snp[[colname[0],colname[3]]]\n",
    "    else:\n",
    "        pval=snp[['RSID',colname[3]]]\n",
    "    pval.columns=['SNP','P']\n",
    "    #pval['N']=N\n",
    "    print(N)\n",
    "    pval.insert(0, 'N', N)\n",
    "    print(pval.head())\n",
    "\n",
    "    pval.to_csv(dir+prefix+'_pval.tsv',sep='\\t',index=False)\n",
    "    print('pval file written to '+dir+prefix+'_pval.tsv')\n",
    "    return pos,pval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "6b0d563a-4239-4509-b414-f930dbc4531b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['facial_hair', 'age_smkinit', 'antisoc', 'friend_sat', 'hr', 'infant_bw', 'LDL', 'maternal_smok', 'townsend', 'age_menarche', 'neurot'])"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_dict.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "89d9513c-a0b9-4e5d-95c4-ad1feadaa3fb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'ex_magma' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Input \u001b[0;32mIn [12]\u001b[0m, in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m t \u001b[38;5;129;01min\u001b[39;00m [\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mLDL\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mmaternal_smok\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtownsend\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mage_menarche\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mneurot\u001b[39m\u001b[38;5;124m'\u001b[39m]:\n\u001b[0;32m----> 2\u001b[0m     \u001b[43mex_magma\u001b[49m(file_dict[t],colname_dict[t],\u001b[38;5;28mint\u001b[39m(meta_data[meta_data[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mlabel\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m==\u001b[39mt][\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mN\u001b[39m\u001b[38;5;124m'\u001b[39m]),t,\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mmagma/\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'ex_magma' is not defined"
     ]
    }
   ],
   "source": [
    "#initial loop for exporting files-\n",
    "#for t in ['LDL', 'maternal_smok', 'townsend', 'age_menarche', 'neurot']:\n",
    "#    pos,pval=ex_magma(file_dict[t],colname_dict[t],int(meta_data[meta_data['label']==t]['N']),t,'magma/')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5e4d68cd-0e02-41c3-8131-755c158b9017",
   "metadata": {},
   "source": [
    "# check output files for failed jobs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "9282fc6e-a2f7-4273-8250-91f72717534c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['magma/infant_bw.genes.annot', 'magma/hr.genes.annot', 'magma/antisoc.genes.annot', 'magma/facial_hair.genes.annot', 'magma/age_menarche.genes.annot', 'magma/neurot.genes.annot', 'magma/townsend.genes.annot', 'magma/friend_sat.genes.annot', 'magma/LDL.genes.annot', 'magma/age_smkinit.genes.annot', 'magma/maternal_smok.genes.annot']\n"
     ]
    }
   ],
   "source": [
    "print(glob.glob(\"magma/*.annot\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "aa349bcf-c11e-4a4d-80cc-f00f2b2fe163",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['infant_bw',\n",
       " 'age_menarche',\n",
       " 'neurot',\n",
       " 'age_smkinit',\n",
       " 'friend_sat',\n",
       " 'hr',\n",
       " 'maternal_smok',\n",
       " 'LDL',\n",
       " 'townsend',\n",
       " 'facial_hair']"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#traits where job succeeded\n",
    "list(map((lambda x: x.split('/')[1].split('.')[0]),glob.glob(\"magma/*.out\")))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "2cc6c219-d5d4-4a85-86b1-6d1e6b5140c3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'antisoc'}\n"
     ]
    }
   ],
   "source": [
    "#traits where job failed\n",
    "rerun=set(map((lambda x: x.split('/')[1].split('.')[0]),glob.glob(\"magma/*.annot\") ))-set(list(map((lambda x: x.split('/')[1].split('.')[0]),glob.glob(\"magma/*.out\") )))\n",
    "print(rerun)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "82e347ac-ea19-439e-a0a8-af3d4ca81011",
   "metadata": {},
   "source": [
    "all initally failed jobs don't have rsids- required formatting apparently, for use of 1000 genomes. \n",
    "none have CHR X and Y- all are numerically ordered"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d1d444dd-8196-4480-8002-6001d30c8df5",
   "metadata": {},
   "source": [
    "only antisoc failed- there is a pvalue that isn't parsable"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "30e7942e-efce-4e91-b7fd-f5b9bded33cf",
   "metadata": {},
   "source": [
    "## merge data dictionary for traits without SNPs with SNP dictionary to add RSIDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "c34c88e9-9b85-40ac-ba96-5ddb13c1950d",
   "metadata": {},
   "outputs": [],
   "source": [
    "snp=pd.read_csv('~/bsl/magma_v1/g1000_eur/g1000_eur.bim',sep='\\t',header=None)\n",
    "snp.columns=['CHR','RSID','CM','BP','A1','A2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "6c04a4cd-70ed-4522-9f9d-3547e39e5213",
   "metadata": {},
   "outputs": [],
   "source": [
    "snp['marker_merge']=(snp['CHR']).map(lambda x:str(x))+':'+(snp['BP']).map(lambda x:str(x))+':'+snp['A1']+':'+snp['A2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "a348131b-2d71-438c-93d3-5df7e80e61aa",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>RSID</th>\n",
       "      <th>CM</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>marker_merge</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>rs537182016</td>\n",
       "      <td>0</td>\n",
       "      <td>10539</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1:10539:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>rs575272151</td>\n",
       "      <td>0</td>\n",
       "      <td>11008</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1:11008:G:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>rs544419019</td>\n",
       "      <td>0</td>\n",
       "      <td>11012</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1:11012:G:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>rs540538026</td>\n",
       "      <td>0</td>\n",
       "      <td>13110</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1:13110:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>rs62635286</td>\n",
       "      <td>0</td>\n",
       "      <td>13116</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1:13116:G:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22665059</th>\n",
       "      <td>23</td>\n",
       "      <td>rs148015245</td>\n",
       "      <td>0</td>\n",
       "      <td>154928909</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>23:154928909:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22665060</th>\n",
       "      <td>23</td>\n",
       "      <td>rs140888996</td>\n",
       "      <td>0</td>\n",
       "      <td>154929192</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>23:154929192:T:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22665061</th>\n",
       "      <td>23</td>\n",
       "      <td>rs557132</td>\n",
       "      <td>0</td>\n",
       "      <td>154929412</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>23:154929412:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22665062</th>\n",
       "      <td>23</td>\n",
       "      <td>rs76850183</td>\n",
       "      <td>0</td>\n",
       "      <td>154929639</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>23:154929639:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22665063</th>\n",
       "      <td>23</td>\n",
       "      <td>rs781880</td>\n",
       "      <td>0</td>\n",
       "      <td>154930230</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>23:154930230:G:A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22665064 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR         RSID  CM         BP A1 A2      marker_merge\n",
       "0           1  rs537182016   0      10539  A  C       1:10539:A:C\n",
       "1           1  rs575272151   0      11008  G  C       1:11008:G:C\n",
       "2           1  rs544419019   0      11012  G  C       1:11012:G:C\n",
       "3           1  rs540538026   0      13110  A  G       1:13110:A:G\n",
       "4           1   rs62635286   0      13116  G  T       1:13116:G:T\n",
       "...       ...          ...  ..        ... .. ..               ...\n",
       "22665059   23  rs148015245   0  154928909  T  C  23:154928909:T:C\n",
       "22665060   23  rs140888996   0  154929192  T  G  23:154929192:T:G\n",
       "22665061   23     rs557132   0  154929412  T  C  23:154929412:T:C\n",
       "22665062   23   rs76850183   0  154929639  C  T  23:154929639:C:T\n",
       "22665063   23     rs781880   0  154930230  G  A  23:154930230:G:A\n",
       "\n",
       "[22665064 rows x 7 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp"
   ]
  },
  {
   "cell_type": "raw",
   "id": "01259afd-17d0-4a27-86c9-6cc50dea8111",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "for t in rerun:\n",
    "    print(t)\n",
    "    file_dict[t]['marker_merge']=file_dict[t][colname_dict[t][1]].map(lambda x:str(x))+':'+file_dict[t][colname_dict[t][2]].map(lambda x:str(x))+':'+file_dict[t][colname_dict[t][4]].str.upper()+':'+file_dict[t][colname_dict[t][5]].str.upper()\n",
    "    file_dict[t]=file_dict[t].merge(snp[['RSID','marker_merge']],on='marker_merge',how='inner')\n",
    "    pos,pval=ex_magma(file_dict[t],colname_dict[t],int(meta_data[meta_data['label']==t]['N']),t,'magma/',True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ad4a978b-b1bf-46e9-913d-83076c02f82b",
   "metadata": {},
   "source": [
    "## fix antisocial error"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "32c5decf-736a-4abb-a60f-746cac1b73a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "t='antisoc'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ea6b6c7-e895-4c7b-9518-a281d303cf9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_dict[t]['marker_merge']=file_dict[t][colname_dict[t][1]].map(lambda x:str(x))+':'+file_dict[t][colname_dict[t][2]].map(lambda x:str(x))+':'+file_dict[t][colname_dict[t][4]].str.upper()+':'+file_dict[t][colname_dict[t][5]].str.upper()\n",
    "file_dict[t]=file_dict[t].merge(snp[['RSID','marker_merge']],on='marker_merge',how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "1d4cd562-9cc2-4bb3-9028-5db96da235ed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>Allele1</th>\n",
       "      <th>Allele2</th>\n",
       "      <th>Weight</th>\n",
       "      <th>Zscore</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Direction</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "      <th>marker_merge</th>\n",
       "      <th>RSID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1283482</th>\n",
       "      <td>14:22262789</td>\n",
       "      <td>t</td>\n",
       "      <td>g</td>\n",
       "      <td>16400.0</td>\n",
       "      <td>0.9190</td>\n",
       "      <td>0.3583</td>\n",
       "      <td>++--+</td>\n",
       "      <td>14</td>\n",
       "      <td>22262789</td>\n",
       "      <td>14:22262789:T:G</td>\n",
       "      <td>rs12886640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1283483</th>\n",
       "      <td>6:50483161</td>\n",
       "      <td>a</td>\n",
       "      <td>c</td>\n",
       "      <td>16400.0</td>\n",
       "      <td>-0.2930</td>\n",
       "      <td>0.7693</td>\n",
       "      <td>--+-+</td>\n",
       "      <td>6</td>\n",
       "      <td>50483161</td>\n",
       "      <td>6:50483161:A:C</td>\n",
       "      <td>rs79400560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1283484</th>\n",
       "      <td>2:134909570</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>1379.0</td>\n",
       "      <td>0.6938</td>\n",
       "      <td>--+++</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>134909570</td>\n",
       "      <td>2:134909570:T:C</td>\n",
       "      <td>rs113396222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1283485</th>\n",
       "      <td>11:126026645</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>16400.0</td>\n",
       "      <td>-0.4780</td>\n",
       "      <td>0.6326</td>\n",
       "      <td>-+-+-</td>\n",
       "      <td>11</td>\n",
       "      <td>126026645</td>\n",
       "      <td>11:126026645:T:C</td>\n",
       "      <td>rs12421177</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           MarkerName Allele1 Allele2   Weight  Zscore P-value Direction CHR  \\\n",
       "1283482   14:22262789       t       g  16400.0  0.9190  0.3583     ++--+  14   \n",
       "1283483    6:50483161       a       c  16400.0 -0.2930  0.7693     --+-+   6   \n",
       "1283484   2:134909570       t       c   1379.0  0.6938   --+++       NaN   2   \n",
       "1283485  11:126026645       t       c  16400.0 -0.4780  0.6326     -+-+-  11   \n",
       "\n",
       "                BP      marker_merge         RSID  \n",
       "1283482   22262789   14:22262789:T:G   rs12886640  \n",
       "1283483   50483161    6:50483161:A:C   rs79400560  \n",
       "1283484  134909570   2:134909570:T:C  rs113396222  \n",
       "1283485  126026645  11:126026645:T:C   rs12421177  "
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "c31575d0-59d7-40d9-b137-cf2b920a34e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#remove line with missing pvalue- and remove any others that may have non-numeric p-values\n",
    "file_dict[t]['P-value']=pd.to_numeric(file_dict[t]['P-value'],'coerce')\n",
    "file_dict[t]=file_dict[t].dropna(subset=['P-value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "03732d36-053f-4413-b621-5e26372eda4b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>Allele1</th>\n",
       "      <th>Allele2</th>\n",
       "      <th>Weight</th>\n",
       "      <th>Zscore</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Direction</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "      <th>marker_merge</th>\n",
       "      <th>RSID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1283482</th>\n",
       "      <td>14:22262789</td>\n",
       "      <td>t</td>\n",
       "      <td>g</td>\n",
       "      <td>16400.0</td>\n",
       "      <td>0.919</td>\n",
       "      <td>0.3583</td>\n",
       "      <td>++--+</td>\n",
       "      <td>14</td>\n",
       "      <td>22262789</td>\n",
       "      <td>14:22262789:T:G</td>\n",
       "      <td>rs12886640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1283483</th>\n",
       "      <td>6:50483161</td>\n",
       "      <td>a</td>\n",
       "      <td>c</td>\n",
       "      <td>16400.0</td>\n",
       "      <td>-0.293</td>\n",
       "      <td>0.7693</td>\n",
       "      <td>--+-+</td>\n",
       "      <td>6</td>\n",
       "      <td>50483161</td>\n",
       "      <td>6:50483161:A:C</td>\n",
       "      <td>rs79400560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1283485</th>\n",
       "      <td>11:126026645</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>16400.0</td>\n",
       "      <td>-0.478</td>\n",
       "      <td>0.6326</td>\n",
       "      <td>-+-+-</td>\n",
       "      <td>11</td>\n",
       "      <td>126026645</td>\n",
       "      <td>11:126026645:T:C</td>\n",
       "      <td>rs12421177</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           MarkerName Allele1 Allele2   Weight  Zscore  P-value Direction CHR  \\\n",
       "1283482   14:22262789       t       g  16400.0   0.919   0.3583     ++--+  14   \n",
       "1283483    6:50483161       a       c  16400.0  -0.293   0.7693     --+-+   6   \n",
       "1283485  11:126026645       t       c  16400.0  -0.478   0.6326     -+-+-  11   \n",
       "\n",
       "                BP      marker_merge        RSID  \n",
       "1283482   22262789   14:22262789:T:G  rs12886640  \n",
       "1283483   50483161    6:50483161:A:C  rs79400560  \n",
       "1283485  126026645  11:126026645:T:C  rs12421177  "
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_dict[t].loc[1283482:1283485] # check line 1283485- one that threw the error- it's missing a pvalue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "b9a2c7ec-7fac-4429-8700-54c0e5af02a6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           SNP CHR        POS\n",
      "0     rs667647   5   29439275\n",
      "1  rs113534962   5   85928892\n",
      "2    rs2366866  10  128341232\n",
      "3     rs472303   3   62707519\n",
      "4   rs11725240   4   55643311\n",
      "pos file written to magma/antisoc_pos.tsv\n",
      "16400\n",
      "       N          SNP       P\n",
      "0  16400     rs667647  0.1959\n",
      "1  16400  rs113534962  0.9342\n",
      "2  16400    rs2366866  0.3463\n",
      "3  16400     rs472303  0.6594\n",
      "4  16400   rs11725240  0.8004\n",
      "pval file written to magma/antisoc_pval.tsv\n"
     ]
    }
   ],
   "source": [
    "pos,pval=ex_magma(file_dict[t],colname_dict[t],int(meta_data[meta_data['label']==t]['N']),t,'magma/',True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2f000818-6cdd-4993-81f7-0604a83c998f",
   "metadata": {},
   "source": [
    "# merge in gene names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "225f8c8b-8907-4746-b1b6-b9cf25ea94dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_loc=pd.read_csv('/tscc/projects/ps-palmer/brittany/magma_v1/NCBI38/NCBI38.gene.loc',sep='\\t',header=None)\n",
    "gene_loc.columns=['ID','CHR','START','STOP','STAND','GENE_NAME']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "6ef57836-ed7b-420b-b11e-f203185cf356",
   "metadata": {},
   "outputs": [],
   "source": [
    "magma_dict={}\n",
    "for t in file_dict.keys():\n",
    "    magma_dict[t]=pd.read_csv('magma/'+t+'.genes.out',delim_whitespace=True)\n",
    "    magma_dict[t]=gene_loc[['ID','GENE_NAME']].merge(magma_dict[t], left_on='ID',right_on='GENE',how='right').reset_index().drop(['index','GENE'],axis=1)\n",
    "    magma_dict[t].columns=['ID', 'GENE', 'CHR', 'START', 'STOP', 'NSNPS', 'NPARAM', 'N','ZSTAT', 'P']\n",
    "    magma_dict[t]['Q']=statsmodels.stats.multitest.fdrcorrection(magma_dict[t]['P'],alpha=0.05,method='indep',is_sorted=False)[1]\n",
    "    magma_dict[t].to_csv('magma/seed_genes/'+t+'_annot.tsv',sep='\\t',index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:env-std-py38]",
   "language": "python",
   "name": "conda-env-env-std-py38-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
