{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3282c56e-1128-432f-9c71-72253eb2bf9d",
   "metadata": {},
   "source": [
    "purpose: run predixcan and FUSION TWAS on externalizing1.0 data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9fd8aeaa-5fbc-4958-a739-59fe4417fedf",
   "metadata": {},
   "source": [
    "# S-PrediXcan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ef205fc-c2c6-4397-9324-cbaf985be3d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH --job-name predixcan_ext\n",
    "#SBATCH --partition condo\n",
    "#SBATCH --qos condo\n",
    "#SBATCH --nodes 1\n",
    "#SBATCH -a 1-13 \n",
    "#SBATCH -c 4\n",
    "#SBATCH -t 2:00:00\n",
    "#SBATCH --mem-per-cpu 8G\n",
    "#SBATCH -o /tscc/nfs/home/bsleger/bsl/SUD_cross_species/job_run_out/predixcan_ext-%j.o\n",
    "#SBATCH -e /tscc/nfs/home/bsleger/bsl/SUD_cross_species/job_run_out/predixcan_ext-%j.e\n",
    "#SBATCH --mail-type END,FAIL\n",
    "#SBATCH --mail-user bsleger@ucsd.edu\n",
    "#SBATCH --account csd795\n",
    "\n",
    "#cd /tscc/projects/ps-palmer/brittany/MetaXcan/GTEx/brain_models/\n",
    "db_list=(`ls *.db`) #len=13 change back for rerun all\n",
    "\n",
    "#failed_rerun=( en_Brain_Caudate_basal_ganglia.db en_Brain_Cerebellar_Hemisphere.db en_Brain_Cerebellum.db en_Brain_Cortex.db en_Brain_Hypothalamus.db en_Brain_Putamen_basal_ganglia.db )\n",
    "\n",
    "cd /tscc/projects/ps-palmer/brittany/MetaXcan/\n",
    "source activate imlabtools\n",
    "\n",
    "m=${db_list[$SLURM_ARRAY_TASK_ID-1]}\n",
    "#m=${failed_rerun[$SLURM_ARRAY_TASK_ID-1]}\n",
    "echo $m\n",
    "\n",
    "software/SPrediXcan.py \\\n",
    "--model_db_path  \"GTEx/brain_models/\"$m \\\n",
    "--covariance GTEx/gtex_v8_expression_elastic_net_snp_smultixcan_covariance.txt.gz \\\n",
    "--gwas_file /tscc/projects/ps-palmer/brittany/SUD_cross_species/ext_sumstat_2019/FINAL.EXT_COMMON_FACTOR.EXTERNALIZING.20191014.PREPARED.wFREQ.A1.txt.gz \\\n",
    "--snp_column SNP \\\n",
    "--effect_allele_column A1 \\\n",
    "--non_effect_allele_column A2 \\\n",
    "--beta_column BETA.A1 \\\n",
    "--pvalue_column P \\\n",
    "--output_file \"results/predixcan_externalizing2019_\"${m##*/}\".csv\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ba9a4364-a37e-4f6a-80fe-3f9aa22b76d5",
   "metadata": {},
   "source": [
    "# FUSION association"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0c94e9c3-1db9-4087-8e2a-c48f33e48807",
   "metadata": {},
   "outputs": [],
   "source": [
    "#decompress all of the GTEx models \n",
    "cd \"${FUS_PATH}/WEIGHTS\"\n",
    "for f in *.tar.bz2; do\n",
    "    tar xjf $f \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e87ea4fe-d986-4dd4-84ef-d70b9ce856aa",
   "metadata": {},
   "source": [
    "The primary input is genome-wide summary statistics in LD-score format. At minimum, this is a flat file with a header row containing the following fields:\n",
    "\n",
    "SNP – SNP identifier (rsID)\n",
    "A1 – first allele (effect allele)\n",
    "A2 – second allele (other allele)\n",
    "Z – Z-scores, sign with respect to A1."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3bdf559c-e733-405c-9367-916a6478e8ec",
   "metadata": {},
   "source": [
    "## run fusion- script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e78aa2a6-59e4-4fd8-8163-ab93106c0aed",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH --job-name ext_FUSION\n",
    "#SBATCH --partition condo\n",
    "#SBATCH --qos condo\n",
    "#SBATCH -a 1-9\n",
    "#SBATCH --time 2:00:00\n",
    "#SBATCH --nodes 1\n",
    "#SBATCH --cpus-per-task 4\n",
    "#SBATCH --mem-per-cpu 4G\n",
    "#SBATCH -o /tscc/nfs/home/bsleger/bsl/SUD_cross_species/job_run_out/ext_FUSION-%j.o\n",
    "#SBATCH -e /tscc/nfs/home/bsleger/bsl/SUD_cross_species/job_run_out/ext_FUSION-%j.e\n",
    "#SBATCH --mail-type END,FAIL\n",
    "#SBATCH --mail-user bsleger@ucsd.edu\n",
    "#SBATCH --account csd795\n",
    "\n",
    "\n",
    "\n",
    "FUS_PATH='/tscc/nfs/home/bsleger/bsl/fusion_twas-master/'\n",
    "SUD_PATH='/tscc/projects/ps-palmer/brittany/SUD_cross_species/'\n",
    "DATA_FILE='ext_sumstat_2019/FINAL.EXT_COMMON_FACTOR.EXTERNALIZING.20191014.PREPARED.wFREQ.A1.txt'\n",
    "OUT_PATH=\"${SUD_PATH}ext_FUSION/\"\n",
    "OUT_PREF=\"ext2019\"\n",
    "\n",
    "\n",
    "cd \"${FUS_PATH}WEIGHTS\"\n",
    "db_list=(`ls GTEx.Brain*.pos`)\n",
    " echo ${#db_list[*]} \n",
    "#db list ls 9 long - make job array that's 1-9\n",
    "\n",
    "cd $FUS_PATH\n",
    "\n",
    "source activate lzenv\n",
    "\n",
    "TISSUE=${db_list[$SLURM_ARRAY_TASK_ID-1]}\n",
    "#m=${db_list[1]}\n",
    "echo $TISSUE\n",
    "\n",
    "\n",
    "for ((CHR = 1; CHR < 23; CHR++));\n",
    "do\n",
    "    echo $CHR\n",
    "    OUT=${OUT_PATH}${OUT_PREF}_${TISSUE}_${CHR}.dat\n",
    "    echo $OUT\n",
    "    Rscript FUSION.assoc_test.R \\\n",
    "    --sumstats $SUD_PATH$DATA_FILE \\\n",
    "    --weights \"./WEIGHTS/\"${TISSUE} \\\n",
    "    --weights_dir ./WEIGHTS/ \\\n",
    "    --ref_ld_chr ./LDREF/1000G.EUR. \\\n",
    "    --chr $CHR \\\n",
    "    --out $OUT\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4b30a70a-8d94-4b4b-940f-8e962c0ac58a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ls: cannot access './LDREF/1000G.EUR.': No such file or directory\n"
     ]
    },
    {
     "ename": "",
     "evalue": "2",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls ./LDREF/1000G.EUR."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "32a10853-eba1-444e-af9c-551198190e61",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GTEx.Brain_Caudate_basal_ganglia.pos\n"
     ]
    }
   ],
   "source": [
    "echo $db_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d785a44-6b79-4396-9d4c-7501b6643cda",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd \"${FUS_PATH}WEIGHTS\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4458e1dc-4cea-46e5-aaac-0cda9c0f90fa",
   "metadata": {},
   "source": [
    "## concat results together"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5ea7819f-9f2a-4519-ac9f-b4bce800a220",
   "metadata": {},
   "source": [
    "### Python- get set of prefixes "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e33a8cfb-6560-417c-a67e-79e3002d86d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a1a33bb6-ab3a-4b29-9bc2-512b51e1ead2",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"/tscc/projects/ps-palmer/brittany/SUD_cross_species/ext_FUSION\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "edf8d92d-6470-4789-ac4f-fa25356f8d18",
   "metadata": {},
   "outputs": [],
   "source": [
    "files=os.listdir()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "56853814-feaa-4f81-a089-e19452adfb7e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "files\n",
    "if 'FUSION_concat' in files:\n",
    "    files.remove('FUSION_concat')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "655325ea-109c-4027-9ab4-507ce8c26561",
   "metadata": {},
   "outputs": [],
   "source": [
    "prefixes=set(map(lambda string: string.split(\".\")[0]+'.'+string.split(\".\")[1], files))\n",
    "if '.ipynb_checkpoints' in prefixes:\n",
    "    prefixes.remove('.ipynb_checkpoints')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "bfac533d-b869-41ad-aea8-38ca19dbb85a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "( 'ext2019_GTEx.Brain_Caudate_basal_ganglia' 'ext2019_GTEx.Brain_Cortex' 'ext2019_GTEx.Brain_Putamen_basal_ganglia' 'ext2019_GTEx.Brain_Hippocampus' 'ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia' 'ext2019_GTEx.Brain_Hypothalamus' 'ext2019_GTEx.Brain_Cerebellum' 'ext2019_GTEx.Brain_Frontal_Cortex_BA9' 'ext2019_GTEx.Brain_Cerebellar_Hemisphere' )\n"
     ]
    }
   ],
   "source": [
    "str='( '\n",
    "for f in prefixes:\n",
    "    str=str+\"'\"+f+\"' \"\n",
    "    \n",
    "str=str[0:len(str)-1]+' )'\n",
    "print(str)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af87de15-9729-401b-bd3c-52ff3b43c068",
   "metadata": {},
   "source": [
    "### bash concat files together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a0c10f9f-a67a-460c-b9a1-e399a0ff1900",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_1.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_2.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_3.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_4.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_5.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_6.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_6.dat.MHC\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_7.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_8.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_9.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_10.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_11.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_12.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_13.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_14.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_15.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_16.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_17.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_18.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_19.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_20.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_21.dat\n",
      "ext2019_GTEx.Brain_Caudate_basal_ganglia.pos_22.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_1.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_2.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_3.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_4.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_5.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_6.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_6.dat.MHC\n",
      "ext2019_GTEx.Brain_Cortex.pos_7.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_8.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_9.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_10.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_11.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_12.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_13.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_14.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_15.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_16.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_17.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_18.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_19.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_20.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_21.dat\n",
      "ext2019_GTEx.Brain_Cortex.pos_22.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_1.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_2.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_3.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_4.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_5.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_6.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_6.dat.MHC\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_7.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_8.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_9.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_10.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_11.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_12.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_13.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_14.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_15.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_16.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_17.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_18.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_19.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_20.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_21.dat\n",
      "ext2019_GTEx.Brain_Putamen_basal_ganglia.pos_22.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_1.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_2.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_3.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_4.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_5.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_6.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_6.dat.MHC\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_7.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_8.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_9.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_10.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_11.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_12.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_13.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_14.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_15.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_16.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_17.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_18.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_19.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_20.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_21.dat\n",
      "ext2019_GTEx.Brain_Hippocampus.pos_22.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_1.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_2.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_3.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_4.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_5.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_6.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_6.dat.MHC\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_7.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_8.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_9.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_10.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_11.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_12.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_13.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_14.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_15.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_16.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_17.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_18.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_19.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_20.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_21.dat\n",
      "ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia.pos_22.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_1.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_2.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_3.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_4.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_5.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_6.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_6.dat.MHC\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_7.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_8.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_9.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_10.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_11.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_12.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_13.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_14.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_15.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_16.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_17.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_18.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_19.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_20.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_21.dat\n",
      "ext2019_GTEx.Brain_Hypothalamus.pos_22.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_1.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_2.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_3.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_4.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_5.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_6.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_6.dat.MHC\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_7.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_8.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_9.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_10.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_11.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_12.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_13.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_14.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_15.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_16.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_17.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_18.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_19.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_20.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_21.dat\n",
      "ext2019_GTEx.Brain_Cerebellum.pos_22.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_1.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_2.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_3.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_4.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_5.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_6.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_6.dat.MHC\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_7.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_8.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_9.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_10.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_11.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_12.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_13.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_14.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_15.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_16.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_17.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_18.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_19.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_20.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_21.dat\n",
      "ext2019_GTEx.Brain_Frontal_Cortex_BA9.pos_22.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_1.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_2.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_3.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_4.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_5.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_6.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_6.dat.MHC\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_7.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_8.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_9.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_10.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_11.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_12.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_13.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_14.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_15.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_16.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_17.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_18.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_19.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_20.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_21.dat\n",
      "ext2019_GTEx.Brain_Cerebellar_Hemisphere.pos_22.dat\n"
     ]
    }
   ],
   "source": [
    "#bash\n",
    "#run to concat all the files \n",
    "cd /tscc/projects/ps-palmer/brittany/SUD_cross_species/ext_FUSION\n",
    "#make directory for concated files\n",
    "if [ ! -d \"FUSION_concat\" ]; then\n",
    "  mkdir FUSION_concat\n",
    "\n",
    "fi\n",
    "\n",
    "prefixes=( 'ext2019_GTEx.Brain_Caudate_basal_ganglia' 'ext2019_GTEx.Brain_Cortex' 'ext2019_GTEx.Brain_Putamen_basal_ganglia' 'ext2019_GTEx.Brain_Hippocampus' 'ext2019_GTEx.Brain_Nucleus_accumbens_basal_ganglia' 'ext2019_GTEx.Brain_Hypothalamus' 'ext2019_GTEx.Brain_Cerebellum' 'ext2019_GTEx.Brain_Frontal_Cortex_BA9' 'ext2019_GTEx.Brain_Cerebellar_Hemisphere' )\n",
    "\n",
    "for p in ${prefixes[@]}; do\n",
    "    for CHR in {1..22}; do\n",
    "        # Define the file\n",
    "        file=\"${p}.pos_${CHR}.dat\"\n",
    "        echo $file\n",
    "        if [ ${CHR} -eq 1 ];\n",
    "        then\n",
    "            cp $file FUSION_concat/${p}.dat\n",
    "        else\n",
    "            awk 'FNR>1' $file >> FUSION_concat/${p}.dat\n",
    "            \n",
    "        fi\n",
    "        if [ $CHR -eq 6 ];\n",
    "        then\n",
    "            file=\"${p}.pos_${CHR}.dat.MHC\"\n",
    "            echo $file\n",
    "            awk 'FNR>1' $file >> FUSION_concat/${p}.dat\n",
    "\n",
    "        fi   \n",
    "    done\n",
    "done\n",
    "\n",
    "# makes files  /tscc/projects/ps-palmer/brittany/SUD_cross_species/nicsa_gwas/results/gwas/mlma_concat/regressedlr_cigday_pc1.mlma\n",
    "# /tscc/projects/ps-palmer/brittany/SUD_cross_species/nicsa_gwas/results/gwas/mlma_concat/regressedlr_cigday_pc1.mlma"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b8a4d090-1c8f-4e9c-a5de-5f5f04afc6be",
   "metadata": {},
   "source": [
    "predixcan and fusion results compared in a different notebook :) \n",
    "\n",
    "SUD_cross_species/scripts/TWAS_FUSION_predixcan_comparison_human_rat.ipynb"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a26734e0-e075-46ae-8fcb-fc8b2a6a451f",
   "metadata": {},
   "source": [
    "## rat fusion"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c348fc21-90d9-4d78-b726-ab24bf8bcbc0",
   "metadata": {},
   "source": [
    "## calculate z-scores rat gwas sumstats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "93230ba3-3d21-49fd-9a13-5788cdae548e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d265f727-a9e6-4761-a21a-3bbaf377ac06",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('/tscc/projects/ps-palmer/brittany/SUD_cross_species/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e3019a69-53ff-4eae-921c-e9fda0c7505a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#t=pd.read_csv('loco_commonFactor/combined_gwas_results_CF_MPH.txt',sep='\\t')\n",
    "t=pd.read_csv('loco_meta/regressedlr_combined_locomotor_sumstat.mlma',sep='\\t')\n",
    "t.columns=t.columns.str.upper()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "72386669-aca4-4596-b383-623c56bc1a9c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHR', 'SNP', 'BP', 'A1', 'A2', 'FREQ', 'B', 'SE', 'P'], dtype='object')"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "t.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2f320be9-2a1d-418d-bb6e-4a75a113b319",
   "metadata": {},
   "outputs": [],
   "source": [
    "t['Z']=t['B']/t['SE']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "4cc7cdec-4c35-4242-ac82-97a0948548a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#t.to_csv('loco_commonFactor/combined_gwas_results_CF_MPH_zscores.txt',sep='\\t')\n",
    "t.to_csv('loco_meta/regressedlr_combined_locomotor_sumstat_zscores.mlma',sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4bb92b41-54e6-42a5-9aee-c4f9c9037fcf",
   "metadata": {},
   "outputs": [],
   "source": [
    "ldref=pd.read_csv('rat_fusion/LDREF/Brain_rn7.9.bim',sep='\\t',header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "7eae56c2-c2a5-4d06-ae57-1d49e71170ab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.9856973828145348"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#check if the SNP IDs match between references\n",
    "len(set(ldref[1]).intersection(t.SNP))/len(set(ldref[1]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "39300372-0706-4768-942c-567c5f8abf31",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:280752</td>\n",
       "      <td>0</td>\n",
       "      <td>280752</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:391998</td>\n",
       "      <td>0</td>\n",
       "      <td>391998</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:392715</td>\n",
       "      <td>0</td>\n",
       "      <td>392715</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:395760</td>\n",
       "      <td>0</td>\n",
       "      <td>395760</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:397990</td>\n",
       "      <td>0</td>\n",
       "      <td>397990</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268757</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:114168818</td>\n",
       "      <td>0</td>\n",
       "      <td>114168818</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268758</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:114170712</td>\n",
       "      <td>0</td>\n",
       "      <td>114170712</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268759</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:114170739</td>\n",
       "      <td>0</td>\n",
       "      <td>114170739</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268760</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:114171690</td>\n",
       "      <td>0</td>\n",
       "      <td>114171690</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268761</th>\n",
       "      <td>9</td>\n",
       "      <td>chr9:114172932</td>\n",
       "      <td>0</td>\n",
       "      <td>114172932</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>268762 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        0               1  2          3  4  5\n",
       "0       9     chr9:280752  0     280752  G  A\n",
       "1       9     chr9:391998  0     391998  C  T\n",
       "2       9     chr9:392715  0     392715  T  G\n",
       "3       9     chr9:395760  0     395760  A  G\n",
       "4       9     chr9:397990  0     397990  A  G\n",
       "...    ..             ... ..        ... .. ..\n",
       "268757  9  chr9:114168818  0  114168818  A  G\n",
       "268758  9  chr9:114170712  0  114170712  T  C\n",
       "268759  9  chr9:114170739  0  114170739  A  G\n",
       "268760  9  chr9:114171690  0  114171690  C  T\n",
       "268761  9  chr9:114172932  0  114172932  A  C\n",
       "\n",
       "[268762 rows x 6 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ldref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "cd3a3c25-124c-4e28-9881-62c5d523f73c",
   "metadata": {},
   "outputs": [],
   "source": [
    "wt=pd.read_csv('rat_fusion/twas-weights-rn7/NAcc/expression.pos',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "836722c6-71c2-41d9-89df-093580ca15cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>WGT</th>\n",
       "      <th>ID</th>\n",
       "      <th>CHR</th>\n",
       "      <th>P0</th>\n",
       "      <th>P1</th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>612</th>\n",
       "      <td>expression/ENSRNOG00000003242.wgt.RDat</td>\n",
       "      <td>ENSRNOG00000003242</td>\n",
       "      <td>9</td>\n",
       "      <td>46622668</td>\n",
       "      <td>46622669</td>\n",
       "      <td>270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693</th>\n",
       "      <td>expression/ENSRNOG00000003825.wgt.RDat</td>\n",
       "      <td>ENSRNOG00000003825</td>\n",
       "      <td>9</td>\n",
       "      <td>47903199</td>\n",
       "      <td>47903200</td>\n",
       "      <td>270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>740</th>\n",
       "      <td>expression/ENSRNOG00000004076.wgt.RDat</td>\n",
       "      <td>ENSRNOG00000004076</td>\n",
       "      <td>9</td>\n",
       "      <td>48253409</td>\n",
       "      <td>48253410</td>\n",
       "      <td>270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>850</th>\n",
       "      <td>expression/ENSRNOG00000004719.wgt.RDat</td>\n",
       "      <td>ENSRNOG00000004719</td>\n",
       "      <td>9</td>\n",
       "      <td>6533638</td>\n",
       "      <td>6533639</td>\n",
       "      <td>270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1737</th>\n",
       "      <td>expression/ENSRNOG00000010174.wgt.RDat</td>\n",
       "      <td>ENSRNOG00000010174</td>\n",
       "      <td>9</td>\n",
       "      <td>16887738</td>\n",
       "      <td>16887739</td>\n",
       "      <td>270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6225</th>\n",
       "      <td>expression/ENSRNOG00000070018.wgt.RDat</td>\n",
       "      <td>ENSRNOG00000070018</td>\n",
       "      <td>9</td>\n",
       "      <td>103827363</td>\n",
       "      <td>103827364</td>\n",
       "      <td>270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6229</th>\n",
       "      <td>expression/ENSRNOG00000070049.wgt.RDat</td>\n",
       "      <td>ENSRNOG00000070049</td>\n",
       "      <td>9</td>\n",
       "      <td>113866731</td>\n",
       "      <td>113866732</td>\n",
       "      <td>270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6247</th>\n",
       "      <td>expression/ENSRNOG00000070300.wgt.RDat</td>\n",
       "      <td>ENSRNOG00000070300</td>\n",
       "      <td>9</td>\n",
       "      <td>113378366</td>\n",
       "      <td>113378367</td>\n",
       "      <td>270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6299</th>\n",
       "      <td>expression/ENSRNOG00000071060.wgt.RDat</td>\n",
       "      <td>ENSRNOG00000071060</td>\n",
       "      <td>9</td>\n",
       "      <td>106093754</td>\n",
       "      <td>106093755</td>\n",
       "      <td>270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6302</th>\n",
       "      <td>expression/ENSRNOG00000071096.wgt.RDat</td>\n",
       "      <td>ENSRNOG00000071096</td>\n",
       "      <td>9</td>\n",
       "      <td>97231745</td>\n",
       "      <td>97231746</td>\n",
       "      <td>270</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>267 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         WGT                  ID  CHR  \\\n",
       "612   expression/ENSRNOG00000003242.wgt.RDat  ENSRNOG00000003242    9   \n",
       "693   expression/ENSRNOG00000003825.wgt.RDat  ENSRNOG00000003825    9   \n",
       "740   expression/ENSRNOG00000004076.wgt.RDat  ENSRNOG00000004076    9   \n",
       "850   expression/ENSRNOG00000004719.wgt.RDat  ENSRNOG00000004719    9   \n",
       "1737  expression/ENSRNOG00000010174.wgt.RDat  ENSRNOG00000010174    9   \n",
       "...                                      ...                 ...  ...   \n",
       "6225  expression/ENSRNOG00000070018.wgt.RDat  ENSRNOG00000070018    9   \n",
       "6229  expression/ENSRNOG00000070049.wgt.RDat  ENSRNOG00000070049    9   \n",
       "6247  expression/ENSRNOG00000070300.wgt.RDat  ENSRNOG00000070300    9   \n",
       "6299  expression/ENSRNOG00000071060.wgt.RDat  ENSRNOG00000071060    9   \n",
       "6302  expression/ENSRNOG00000071096.wgt.RDat  ENSRNOG00000071096    9   \n",
       "\n",
       "             P0         P1    N  \n",
       "612    46622668   46622669  270  \n",
       "693    47903199   47903200  270  \n",
       "740    48253409   48253410  270  \n",
       "850     6533638    6533639  270  \n",
       "1737   16887738   16887739  270  \n",
       "...         ...        ...  ...  \n",
       "6225  103827363  103827364  270  \n",
       "6229  113866731  113866732  270  \n",
       "6247  113378366  113378367  270  \n",
       "6299  106093754  106093755  270  \n",
       "6302   97231745   97231746  270  \n",
       "\n",
       "[267 rows x 6 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wt[wt.CHR==9]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a61893df-c8f3-4afa-972f-ea95b2ccdebe",
   "metadata": {},
   "source": [
    "## run fusion- bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "2815488a-95dc-45e3-9fd1-2cccf8672c03",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda activate lzenv\n",
    " \n",
    "TISSUE='NAcc2'\n",
    "VAR_TYPE='expression'\n",
    "\n",
    "FUS_PATH='/tscc/nfs/home/bsleger/bsl/fusion_twas-master/'\n",
    "\n",
    "cd $FUS_PATH\n",
    "SUD_PATH='/tscc/projects/ps-palmer/brittany/SUD_cross_species/'\n",
    "OUT_PATH=\"${SUD_PATH}rat_fusion/output/\"\n",
    "OUT_PREF='loco_meta'\n",
    "DATA_FILE='loco_meta/regressedlr_combined_locomotor_sumstat_zscores.mlma'\n",
    "FUS_REF_PATH=${SUD_PATH}'rat_fusion/'\n",
    "WEIGHTS_PATH=$FUS_REF_PATH\"twas-weights-rn7/\"${TISSUE}\n",
    "\n",
    "for ((CHR = 1; CHR < 22; CHR++)); do\n",
    "    OUT=${OUT_PATH}${OUT_PREF}_${TISSUE}_${CHR}.dat\n",
    "    echo $OUT\n",
    "    Rscript FUSION.assoc_test.R \\\n",
    "    --sumstats $SUD_PATH$DATA_FILE \\\n",
    "    --weights ${WEIGHTS_PATH}/${VAR_TYPE}'.pos' \\\n",
    "    --weights_dir $WEIGHTS_PATH \\\n",
    "    --ref_ld_chr $FUS_REF_PATH/LDREF/Brain_rn7. \\\n",
    "    --chr $CHR \\\n",
    "    --out $OUT\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c8e197b-2fd2-40e0-b19f-563366d42d03",
   "metadata": {},
   "outputs": [],
   "source": [
    "#decompress input files\n",
    "\n",
    "#tissues=( 'Adipose'  'BLA'  'Brain'  'Eye'  'IL'  'LHb'  'Liver'  'NAcc'  'NAcc1'  'NAcc2'  'OFC'  'PL'  'PL1'  'PL2' )\n",
    "tissues=( 'Adipose'  'BLA'  'Brain'  'Eye'  'IL'  'LHb'  'Liver' 'OFC'  'PL'  'PL1'  'PL2' )\n",
    "for d in ${tissues[@]}; do    \n",
    "echo $d\n",
    "cd $d\n",
    "    for f in *.tar.bz2; do\n",
    "        tar xjf $f \n",
    "    done\n",
    "cd ..\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "c8b9e6e7-f470-4b37-b386-96d7ccc85163",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash: Rscript: command not found\n"
     ]
    },
    {
     "ename": "",
     "evalue": "127",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "for ((CHR = 1; CHR < 22; CHR++)); do\n",
    "    OUT=${OUT_PATH}${OUT_PREF}_${TISSUE}_${CHR}.dat\n",
    "    echo $OUT\n",
    "    Rscript FUSION.assoc_test.R \\\n",
    "    --sumstats $SUD_PATH$DATA_FILE \\\n",
    "    --weights ${WEIGHTS_PATH}/${VAR_TYPE}'.pos' \\\n",
    "    --weights_dir $WEIGHTS_PATH \\\n",
    "    --ref_ld_chr $FUS_REF_PATH/LDREF/Brain_rn7. \\\n",
    "    --chr $CHR \\\n",
    "    --out $OUT\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "17eb1a84-a522-4cfe-b4fe-5e1096633d6e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "loco_gsem_NAcc_1.dat\n",
      "loco_gsem_NAcc_2.dat\n",
      "loco_gsem_NAcc_3.dat\n",
      "loco_gsem_NAcc_4.dat\n",
      "loco_gsem_NAcc_5.dat\n",
      "loco_gsem_NAcc_6.dat\n",
      "loco_gsem_NAcc_6.dat.MHC\n",
      "loco_gsem_NAcc_7.dat\n",
      "loco_gsem_NAcc_8.dat\n",
      "loco_gsem_NAcc_9.dat\n",
      "loco_gsem_NAcc_10.dat\n",
      "loco_gsem_NAcc_11.dat\n",
      "loco_gsem_NAcc_12.dat\n",
      "loco_gsem_NAcc_13.dat\n",
      "loco_gsem_NAcc_14.dat\n",
      "loco_gsem_NAcc_15.dat\n",
      "loco_gsem_NAcc_16.dat\n",
      "loco_gsem_NAcc_17.dat\n",
      "loco_gsem_NAcc_18.dat\n",
      "loco_gsem_NAcc_19.dat\n",
      "loco_gsem_NAcc_20.dat\n",
      "loco_gsem_NAcc_21.dat\n",
      "awk: fatal: cannot open file `loco_gsem_NAcc_21.dat' for reading: No such file or directory\n",
      "loco_gsem_NAcc_22.dat\n",
      "awk: fatal: cannot open file `loco_gsem_NAcc_22.dat' for reading: No such file or directory\n",
      "loco_gsem_NAcc1_1.dat\n",
      "loco_gsem_NAcc1_2.dat\n",
      "loco_gsem_NAcc1_3.dat\n",
      "loco_gsem_NAcc1_4.dat\n",
      "loco_gsem_NAcc1_5.dat\n",
      "loco_gsem_NAcc1_6.dat\n",
      "loco_gsem_NAcc1_6.dat.MHC\n",
      "loco_gsem_NAcc1_7.dat\n",
      "loco_gsem_NAcc1_8.dat\n",
      "loco_gsem_NAcc1_9.dat\n",
      "loco_gsem_NAcc1_10.dat\n",
      "loco_gsem_NAcc1_11.dat\n",
      "loco_gsem_NAcc1_12.dat\n",
      "loco_gsem_NAcc1_13.dat\n",
      "loco_gsem_NAcc1_14.dat\n",
      "loco_gsem_NAcc1_15.dat\n",
      "loco_gsem_NAcc1_16.dat\n",
      "loco_gsem_NAcc1_17.dat\n",
      "loco_gsem_NAcc1_18.dat\n",
      "loco_gsem_NAcc1_19.dat\n",
      "loco_gsem_NAcc1_20.dat\n",
      "loco_gsem_NAcc1_21.dat\n",
      "awk: fatal: cannot open file `loco_gsem_NAcc1_21.dat' for reading: No such file or directory\n",
      "loco_gsem_NAcc1_22.dat\n",
      "awk: fatal: cannot open file `loco_gsem_NAcc1_22.dat' for reading: No such file or directory\n",
      "loco_gsem_NAcc2_1.dat\n",
      "loco_gsem_NAcc2_2.dat\n",
      "loco_gsem_NAcc2_3.dat\n",
      "loco_gsem_NAcc2_4.dat\n",
      "loco_gsem_NAcc2_5.dat\n",
      "loco_gsem_NAcc2_6.dat\n",
      "loco_gsem_NAcc2_6.dat.MHC\n",
      "loco_gsem_NAcc2_7.dat\n",
      "loco_gsem_NAcc2_8.dat\n",
      "loco_gsem_NAcc2_9.dat\n",
      "loco_gsem_NAcc2_10.dat\n",
      "loco_gsem_NAcc2_11.dat\n",
      "loco_gsem_NAcc2_12.dat\n",
      "loco_gsem_NAcc2_13.dat\n",
      "loco_gsem_NAcc2_14.dat\n",
      "loco_gsem_NAcc2_15.dat\n",
      "loco_gsem_NAcc2_16.dat\n",
      "loco_gsem_NAcc2_17.dat\n",
      "loco_gsem_NAcc2_18.dat\n",
      "loco_gsem_NAcc2_19.dat\n",
      "loco_gsem_NAcc2_20.dat\n",
      "loco_gsem_NAcc2_21.dat\n",
      "awk: fatal: cannot open file `loco_gsem_NAcc2_21.dat' for reading: No such file or directory\n",
      "loco_gsem_NAcc2_22.dat\n",
      "awk: fatal: cannot open file `loco_gsem_NAcc2_22.dat' for reading: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "#bash\n",
    "#run to concat all the files \n",
    "cd /tscc/projects/ps-palmer/brittany/SUD_cross_species/rat_fusion/output\n",
    "#make directory for concated files\n",
    "if [ ! -d \"FUSION_concat\" ]; then\n",
    "  mkdir FUSION_concat\n",
    "\n",
    "fi\n",
    "\n",
    "\n",
    "prefixes=( 'loco_gsem_NAcc' 'loco_gsem_NAcc1' 'loco_gsem_NAcc2' )\n",
    "\n",
    "for p in ${prefixes[@]}; do\n",
    "    for CHR in {1..20}; do\n",
    "        # Define the file\n",
    "        file=\"${p}_${CHR}.dat\"\n",
    "        echo $file\n",
    "        if [ ${CHR} -eq 1 ];\n",
    "        then\n",
    "            cp $file FUSION_concat/${p}.dat\n",
    "        else\n",
    "            awk 'FNR>1' $file >> FUSION_concat/${p}.dat\n",
    "            \n",
    "        fi\n",
    "        if [ $CHR -eq 6 ];\n",
    "        then\n",
    "            file=\"${p}_${CHR}.dat.MHC\"\n",
    "            echo $file\n",
    "            awk 'FNR>1' $file >> FUSION_concat/${p}.dat\n",
    "\n",
    "        fi   \n",
    "    done\n",
    "done\n",
    "\n",
    "# makes files  /tscc/projects/ps-palmer/brittany/SUD_cross_species/nicsa_gwas/results/gwas/mlma_concat/regressedlr_cigday_pc1.mlma\n",
    "# /tscc/projects/ps-palmer/brittany/SUD_cross_species/nicsa_gwas/results/gwas/mlma_concat/regressedlr_cigday_pc1.mlma"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:env-std-py38]",
   "language": "python",
   "name": "conda-env-env-std-py38-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
